{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"provenance":[],"authorship_tag":"ABX9TyMyfzls58SASImwGNt+nXMW"},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"code","execution_count":1,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"EQgdunbD3nvI","executionInfo":{"status":"ok","timestamp":1750030211807,"user_tz":-540,"elapsed":19246,"user":{"displayName":"권혜빈","userId":"01947391368488800076"}},"outputId":"6f4fb121-7c8d-4a30-ccc7-b7aaf5c8f200"},"outputs":[{"output_type":"stream","name":"stdout","text":["Mounted at /content/drive\n"]}],"source":["from google.colab import drive\n","drive.mount('/content/drive')"]},{"cell_type":"markdown","source":["# Expression_df 데이터프레임"],"metadata":{"id":"K0JhMLfVDYFE"}},{"cell_type":"code","source":["# 압축 해제\n","import gzip\n","import shutil\n","\n","with gzip.open(\"/content/drive/MyDrive/BIO_MEDICAL_PROJECT/obesity_gene_analysis/data/GSE77962_series_matrix.txt.gz\", 'rb') as f_in:\n","    with open(\"/content/drive/MyDrive/BIO_MEDICAL_PROJECT/obesity_gene_analysis/data/GSE77962_series_matrix.txt\", 'wb') as f_out:\n","        shutil.copyfileobj(f_in, f_out)"],"metadata":{"id":"Fg-Bm3907XEU","executionInfo":{"status":"ok","timestamp":1750030359032,"user_tz":-540,"elapsed":5075,"user":{"displayName":"권혜빈","userId":"01947391368488800076"}}},"execution_count":2,"outputs":[]},{"cell_type":"code","source":["import pandas as pd\n","GSE77962_series_matrix_df = pd.read_csv(\"/content/drive/MyDrive/BIO_MEDICAL_PROJECT/obesity_gene_analysis/data/GSE77962_series_matrix.txt\", sep=\"\\t\", comment='!')"],"metadata":{"id":"ZNGsIfaU8AVA","executionInfo":{"status":"ok","timestamp":1750030446696,"user_tz":-540,"elapsed":1110,"user":{"displayName":"권혜빈","userId":"01947391368488800076"}}},"execution_count":6,"outputs":[]},{"cell_type":"code","source":["GSE77962_series_matrix_df"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":443},"id":"T4pH-sY68LC1","executionInfo":{"status":"ok","timestamp":1750033304804,"user_tz":-540,"elapsed":91,"user":{"displayName":"권혜빈","userId":"01947391368488800076"}},"outputId":"e163b7fe-d904-4a5a-a619-79b57d88456f"},"execution_count":19,"outputs":[{"output_type":"execute_result","data":{"text/plain":["        ID_REF  GSM2062466  GSM2062467  GSM2062468  GSM2062469  GSM2062470  \\\n","0      7892501    4.428059    4.175376    4.025431    4.058749    5.077221   \n","1      7892502    3.015303    3.382805    4.266405    3.859811    3.524342   \n","2      7892503    2.469030    1.683800    1.353440    1.833336    1.773266   \n","3      7892504    7.469548    7.575583    8.319473    7.877318    7.463798   \n","4      7892505    2.655132    1.349682    1.610853    1.264390    2.165488   \n","...        ...         ...         ...         ...         ...         ...   \n","33292  8180414    3.771268    3.827958    4.060053    4.299973    4.432689   \n","33293  8180415    7.106688    6.772125    6.657147    7.350512    7.153225   \n","33294  8180416    3.159829    3.384084    3.988990    3.447247    3.426313   \n","33295  8180417    7.572312    7.260562    7.102751    7.543385    7.276615   \n","33296  8180418    8.086955    7.506301    7.384701    7.924522    7.381025   \n","\n","       GSM2062471  GSM2062472  GSM2062473  GSM2062474  ...  GSM2062608  \\\n","0        4.167342    5.217643    5.940471    4.454951  ...    4.535857   \n","1        3.983906    3.177080    3.634795    2.864439  ...    4.340592   \n","2        2.073515    2.035071    1.663281    1.647068  ...    2.233719   \n","3        7.777693    8.129641    7.505768    8.378240  ...    8.569401   \n","4        1.744837    2.018746    1.284328    1.381185  ...    1.300862   \n","...           ...         ...         ...         ...  ...         ...   \n","33292    4.450297    4.603566    4.273023    4.684982  ...    4.710916   \n","33293    7.043244    7.231078    6.428579    6.299577  ...    6.784532   \n","33294    3.424887    3.423193    3.636731    3.205910  ...    3.607290   \n","33295    7.051876    7.267355    6.877710    6.694160  ...    7.217937   \n","33296    7.339969    7.359909    7.207605    6.963106  ...    7.129784   \n","\n","       GSM2062609  GSM2062610  GSM2062611  GSM2062612  GSM2062613  GSM2062614  \\\n","0        4.908356    5.564917    2.662255    5.434389    4.318741    5.142372   \n","1        3.783674    4.117283    3.957558    3.836091    3.138714    4.347246   \n","2        2.339964    2.191594    2.925588    1.825886    2.154871    2.842330   \n","3        8.639623    8.087239    8.292656    7.683577    7.317489    7.908107   \n","4        1.628132    1.477364    1.657482    1.028516    1.138573    1.237365   \n","...           ...         ...         ...         ...         ...         ...   \n","33292    4.237877    4.360936    4.887058    4.566632    4.723105    4.759401   \n","33293    6.962716    6.634560    6.862306    6.961862    6.817866    6.760013   \n","33294    3.704066    3.750592    3.502732    3.150830    3.621640    3.182532   \n","33295    6.992108    7.343185    6.941972    7.337702    7.330790    7.255967   \n","33296    6.373951    6.973372    6.545439    7.355312    6.848314    7.308610   \n","\n","       GSM2062615  GSM2062616  GSM2062617  \n","0        5.739793    5.183112    5.107738  \n","1        4.117830    4.315688    3.758828  \n","2        2.123143    2.222103    1.485676  \n","3        7.973787    7.984416    7.821190  \n","4        1.726360    1.865871    1.279610  \n","...           ...         ...         ...  \n","33292    5.170525    4.519322    3.996389  \n","33293    6.856684    7.077445    7.054198  \n","33294    3.449176    3.040495    3.080801  \n","33295    7.227201    7.187702    7.173843  \n","33296    6.973893    7.526776    6.731832  \n","\n","[33297 rows x 153 columns]"],"text/html":["\n","  <div id=\"df-d8112210-9045-45e9-a9cf-abcc7f7a5d9d\" class=\"colab-df-container\">\n","    <div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>ID_REF</th>\n","      <th>GSM2062466</th>\n","      <th>GSM2062467</th>\n","      <th>GSM2062468</th>\n","      <th>GSM2062469</th>\n","      <th>GSM2062470</th>\n","      <th>GSM2062471</th>\n","      <th>GSM2062472</th>\n","      <th>GSM2062473</th>\n","      <th>GSM2062474</th>\n","      <th>...</th>\n","      <th>GSM2062608</th>\n","      <th>GSM2062609</th>\n","      <th>GSM2062610</th>\n","      <th>GSM2062611</th>\n","      <th>GSM2062612</th>\n","      <th>GSM2062613</th>\n","      <th>GSM2062614</th>\n","      <th>GSM2062615</th>\n","      <th>GSM2062616</th>\n","      <th>GSM2062617</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>7892501</td>\n","      <td>4.428059</td>\n","      <td>4.175376</td>\n","      <td>4.025431</td>\n","      <td>4.058749</td>\n","      <td>5.077221</td>\n","      <td>4.167342</td>\n","      <td>5.217643</td>\n","      <td>5.940471</td>\n","      <td>4.454951</td>\n","      <td>...</td>\n","      <td>4.535857</td>\n","      <td>4.908356</td>\n","      <td>5.564917</td>\n","      <td>2.662255</td>\n","      <td>5.434389</td>\n","      <td>4.318741</td>\n","      <td>5.142372</td>\n","      <td>5.739793</td>\n","      <td>5.183112</td>\n","      <td>5.107738</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>7892502</td>\n","      <td>3.015303</td>\n","      <td>3.382805</td>\n","      <td>4.266405</td>\n","      <td>3.859811</td>\n","      <td>3.524342</td>\n","      <td>3.983906</td>\n","      <td>3.177080</td>\n","      <td>3.634795</td>\n","      <td>2.864439</td>\n","      <td>...</td>\n","      <td>4.340592</td>\n","      <td>3.783674</td>\n","      <td>4.117283</td>\n","      <td>3.957558</td>\n","      <td>3.836091</td>\n","      <td>3.138714</td>\n","      <td>4.347246</td>\n","      <td>4.117830</td>\n","      <td>4.315688</td>\n","      <td>3.758828</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>7892503</td>\n","      <td>2.469030</td>\n","      <td>1.683800</td>\n","      <td>1.353440</td>\n","      <td>1.833336</td>\n","      <td>1.773266</td>\n","      <td>2.073515</td>\n","      <td>2.035071</td>\n","      <td>1.663281</td>\n","      <td>1.647068</td>\n","      <td>...</td>\n","      <td>2.233719</td>\n","      <td>2.339964</td>\n","      <td>2.191594</td>\n","      <td>2.925588</td>\n","      <td>1.825886</td>\n","      <td>2.154871</td>\n","      <td>2.842330</td>\n","      <td>2.123143</td>\n","      <td>2.222103</td>\n","      <td>1.485676</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>7892504</td>\n","      <td>7.469548</td>\n","      <td>7.575583</td>\n","      <td>8.319473</td>\n","      <td>7.877318</td>\n","      <td>7.463798</td>\n","      <td>7.777693</td>\n","      <td>8.129641</td>\n","      <td>7.505768</td>\n","      <td>8.378240</td>\n","      <td>...</td>\n","      <td>8.569401</td>\n","      <td>8.639623</td>\n","      <td>8.087239</td>\n","      <td>8.292656</td>\n","      <td>7.683577</td>\n","      <td>7.317489</td>\n","      <td>7.908107</td>\n","      <td>7.973787</td>\n","      <td>7.984416</td>\n","      <td>7.821190</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>7892505</td>\n","      <td>2.655132</td>\n","      <td>1.349682</td>\n","      <td>1.610853</td>\n","      <td>1.264390</td>\n","      <td>2.165488</td>\n","      <td>1.744837</td>\n","      <td>2.018746</td>\n","      <td>1.284328</td>\n","      <td>1.381185</td>\n","      <td>...</td>\n","      <td>1.300862</td>\n","      <td>1.628132</td>\n","      <td>1.477364</td>\n","      <td>1.657482</td>\n","      <td>1.028516</td>\n","      <td>1.138573</td>\n","      <td>1.237365</td>\n","      <td>1.726360</td>\n","      <td>1.865871</td>\n","      <td>1.279610</td>\n","    </tr>\n","    <tr>\n","      <th>...</th>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","    </tr>\n","    <tr>\n","      <th>33292</th>\n","      <td>8180414</td>\n","      <td>3.771268</td>\n","      <td>3.827958</td>\n","      <td>4.060053</td>\n","      <td>4.299973</td>\n","      <td>4.432689</td>\n","      <td>4.450297</td>\n","      <td>4.603566</td>\n","      <td>4.273023</td>\n","      <td>4.684982</td>\n","      <td>...</td>\n","      <td>4.710916</td>\n","      <td>4.237877</td>\n","      <td>4.360936</td>\n","      <td>4.887058</td>\n","      <td>4.566632</td>\n","      <td>4.723105</td>\n","      <td>4.759401</td>\n","      <td>5.170525</td>\n","      <td>4.519322</td>\n","      <td>3.996389</td>\n","    </tr>\n","    <tr>\n","      <th>33293</th>\n","      <td>8180415</td>\n","      <td>7.106688</td>\n","      <td>6.772125</td>\n","      <td>6.657147</td>\n","      <td>7.350512</td>\n","      <td>7.153225</td>\n","      <td>7.043244</td>\n","      <td>7.231078</td>\n","      <td>6.428579</td>\n","      <td>6.299577</td>\n","      <td>...</td>\n","      <td>6.784532</td>\n","      <td>6.962716</td>\n","      <td>6.634560</td>\n","      <td>6.862306</td>\n","      <td>6.961862</td>\n","      <td>6.817866</td>\n","      <td>6.760013</td>\n","      <td>6.856684</td>\n","      <td>7.077445</td>\n","      <td>7.054198</td>\n","    </tr>\n","    <tr>\n","      <th>33294</th>\n","      <td>8180416</td>\n","      <td>3.159829</td>\n","      <td>3.384084</td>\n","      <td>3.988990</td>\n","      <td>3.447247</td>\n","      <td>3.426313</td>\n","      <td>3.424887</td>\n","      <td>3.423193</td>\n","      <td>3.636731</td>\n","      <td>3.205910</td>\n","      <td>...</td>\n","      <td>3.607290</td>\n","      <td>3.704066</td>\n","      <td>3.750592</td>\n","      <td>3.502732</td>\n","      <td>3.150830</td>\n","      <td>3.621640</td>\n","      <td>3.182532</td>\n","      <td>3.449176</td>\n","      <td>3.040495</td>\n","      <td>3.080801</td>\n","    </tr>\n","    <tr>\n","      <th>33295</th>\n","      <td>8180417</td>\n","      <td>7.572312</td>\n","      <td>7.260562</td>\n","      <td>7.102751</td>\n","      <td>7.543385</td>\n","      <td>7.276615</td>\n","      <td>7.051876</td>\n","      <td>7.267355</td>\n","      <td>6.877710</td>\n","      <td>6.694160</td>\n","      <td>...</td>\n","      <td>7.217937</td>\n","      <td>6.992108</td>\n","      <td>7.343185</td>\n","      <td>6.941972</td>\n","      <td>7.337702</td>\n","      <td>7.330790</td>\n","      <td>7.255967</td>\n","      <td>7.227201</td>\n","      <td>7.187702</td>\n","      <td>7.173843</td>\n","    </tr>\n","    <tr>\n","      <th>33296</th>\n","      <td>8180418</td>\n","      <td>8.086955</td>\n","      <td>7.506301</td>\n","      <td>7.384701</td>\n","      <td>7.924522</td>\n","      <td>7.381025</td>\n","      <td>7.339969</td>\n","      <td>7.359909</td>\n","      <td>7.207605</td>\n","      <td>6.963106</td>\n","      <td>...</td>\n","      <td>7.129784</td>\n","      <td>6.373951</td>\n","      <td>6.973372</td>\n","      <td>6.545439</td>\n","      <td>7.355312</td>\n","      <td>6.848314</td>\n","      <td>7.308610</td>\n","      <td>6.973893</td>\n","      <td>7.526776</td>\n","      <td>6.731832</td>\n","    </tr>\n","  </tbody>\n","</table>\n","<p>33297 rows × 153 columns</p>\n","</div>\n","    <div class=\"colab-df-buttons\">\n","\n","  <div class=\"colab-df-container\">\n","    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-d8112210-9045-45e9-a9cf-abcc7f7a5d9d')\"\n","            title=\"Convert this dataframe to an interactive table.\"\n","            style=\"display:none;\">\n","\n","  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n","    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n","  </svg>\n","    </button>\n","\n","  <style>\n","    .colab-df-container {\n","      display:flex;\n","      gap: 12px;\n","    }\n","\n","    .colab-df-convert {\n","      background-color: #E8F0FE;\n","      border: none;\n","      border-radius: 50%;\n","      cursor: pointer;\n","      display: none;\n","      fill: #1967D2;\n","      height: 32px;\n","      padding: 0 0 0 0;\n","      width: 32px;\n","    }\n","\n","    .colab-df-convert:hover {\n","      background-color: #E2EBFA;\n","      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n","      fill: #174EA6;\n","    }\n","\n","    .colab-df-buttons div {\n","      margin-bottom: 4px;\n","    }\n","\n","    [theme=dark] .colab-df-convert {\n","      background-color: #3B4455;\n","      fill: #D2E3FC;\n","    }\n","\n","    [theme=dark] .colab-df-convert:hover {\n","      background-color: #434B5C;\n","      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n","      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n","      fill: #FFFFFF;\n","    }\n","  </style>\n","\n","    <script>\n","      const buttonEl =\n","        document.querySelector('#df-d8112210-9045-45e9-a9cf-abcc7f7a5d9d button.colab-df-convert');\n","      buttonEl.style.display =\n","        google.colab.kernel.accessAllowed ? 'block' : 'none';\n","\n","      async function convertToInteractive(key) {\n","        const element = document.querySelector('#df-d8112210-9045-45e9-a9cf-abcc7f7a5d9d');\n","        const dataTable =\n","          await google.colab.kernel.invokeFunction('convertToInteractive',\n","                                                    [key], {});\n","        if (!dataTable) return;\n","\n","        const docLinkHtml = 'Like what you see? Visit the ' +\n","          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n","          + ' to learn more about interactive tables.';\n","        element.innerHTML = '';\n","        dataTable['output_type'] = 'display_data';\n","        await google.colab.output.renderOutput(dataTable, element);\n","        const docLink = document.createElement('div');\n","        docLink.innerHTML = docLinkHtml;\n","        element.appendChild(docLink);\n","      }\n","    </script>\n","  </div>\n","\n","\n","    <div id=\"df-a6246bc6-ddf6-4898-ab6c-83a23ade7f4a\">\n","      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-a6246bc6-ddf6-4898-ab6c-83a23ade7f4a')\"\n","                title=\"Suggest charts\"\n","                style=\"display:none;\">\n","\n","<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n","     width=\"24px\">\n","    <g>\n","        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n","    </g>\n","</svg>\n","      </button>\n","\n","<style>\n","  .colab-df-quickchart {\n","      --bg-color: #E8F0FE;\n","      --fill-color: #1967D2;\n","      --hover-bg-color: #E2EBFA;\n","      --hover-fill-color: #174EA6;\n","      --disabled-fill-color: #AAA;\n","      --disabled-bg-color: #DDD;\n","  }\n","\n","  [theme=dark] .colab-df-quickchart {\n","      --bg-color: #3B4455;\n","      --fill-color: #D2E3FC;\n","      --hover-bg-color: #434B5C;\n","      --hover-fill-color: #FFFFFF;\n","      --disabled-bg-color: #3B4455;\n","      --disabled-fill-color: #666;\n","  }\n","\n","  .colab-df-quickchart {\n","    background-color: var(--bg-color);\n","    border: none;\n","    border-radius: 50%;\n","    cursor: pointer;\n","    display: none;\n","    fill: var(--fill-color);\n","    height: 32px;\n","    padding: 0;\n","    width: 32px;\n","  }\n","\n","  .colab-df-quickchart:hover {\n","    background-color: var(--hover-bg-color);\n","    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n","    fill: var(--button-hover-fill-color);\n","  }\n","\n","  .colab-df-quickchart-complete:disabled,\n","  .colab-df-quickchart-complete:disabled:hover {\n","    background-color: var(--disabled-bg-color);\n","    fill: var(--disabled-fill-color);\n","    box-shadow: none;\n","  }\n","\n","  .colab-df-spinner {\n","    border: 2px solid var(--fill-color);\n","    border-color: transparent;\n","    border-bottom-color: var(--fill-color);\n","    animation:\n","      spin 1s steps(1) infinite;\n","  }\n","\n","  @keyframes spin {\n","    0% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","      border-left-color: var(--fill-color);\n","    }\n","    20% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    30% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","      border-right-color: var(--fill-color);\n","    }\n","    40% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    60% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","    }\n","    80% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-bottom-color: var(--fill-color);\n","    }\n","    90% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","    }\n","  }\n","</style>\n","\n","      <script>\n","        async function quickchart(key) {\n","          const quickchartButtonEl =\n","            document.querySelector('#' + key + ' button');\n","          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n","          quickchartButtonEl.classList.add('colab-df-spinner');\n","          try {\n","            const charts = await google.colab.kernel.invokeFunction(\n","                'suggestCharts', [key], {});\n","          } catch (error) {\n","            console.error('Error during call to suggestCharts:', error);\n","          }\n","          quickchartButtonEl.classList.remove('colab-df-spinner');\n","          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n","        }\n","        (() => {\n","          let quickchartButtonEl =\n","            document.querySelector('#df-a6246bc6-ddf6-4898-ab6c-83a23ade7f4a button');\n","          quickchartButtonEl.style.display =\n","            google.colab.kernel.accessAllowed ? 'block' : 'none';\n","        })();\n","      </script>\n","    </div>\n","\n","  <div id=\"id_0bb5b736-f9be-4798-8e00-3d8108d9e647\">\n","    <style>\n","      .colab-df-generate {\n","        background-color: #E8F0FE;\n","        border: none;\n","        border-radius: 50%;\n","        cursor: pointer;\n","        display: none;\n","        fill: #1967D2;\n","        height: 32px;\n","        padding: 0 0 0 0;\n","        width: 32px;\n","      }\n","\n","      .colab-df-generate:hover {\n","        background-color: #E2EBFA;\n","        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n","        fill: #174EA6;\n","      }\n","\n","      [theme=dark] .colab-df-generate {\n","        background-color: #3B4455;\n","        fill: #D2E3FC;\n","      }\n","\n","      [theme=dark] .colab-df-generate:hover {\n","        background-color: #434B5C;\n","        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n","        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n","        fill: #FFFFFF;\n","      }\n","    </style>\n","    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('GSE77962_series_matrix_df')\"\n","            title=\"Generate code using this dataframe.\"\n","            style=\"display:none;\">\n","\n","  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n","       width=\"24px\">\n","    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n","  </svg>\n","    </button>\n","    <script>\n","      (() => {\n","      const buttonEl =\n","        document.querySelector('#id_0bb5b736-f9be-4798-8e00-3d8108d9e647 button.colab-df-generate');\n","      buttonEl.style.display =\n","        google.colab.kernel.accessAllowed ? 'block' : 'none';\n","\n","      buttonEl.onclick = () => {\n","        google.colab.notebook.generateWithVariable('GSE77962_series_matrix_df');\n","      }\n","      })();\n","    </script>\n","  </div>\n","\n","    </div>\n","  </div>\n"],"application/vnd.google.colaboratory.intrinsic+json":{"type":"dataframe","variable_name":"GSE77962_series_matrix_df"}},"metadata":{},"execution_count":19}]},{"cell_type":"markdown","source":["| 항목             | 의미                                                          |\n","| -------------- | ----------------------------------------------------------- |\n","| **행 (33297개)** | 각각 하나의 **유전자 프로브** (ID\\_REF). 즉, 유전자 표현형에 대응하는 탐침(probe) ID |\n","| **열 (153개)**   | 각각 하나의 **샘플 (GSM ID)** = 피실험자들의 **조직 샘플 (예: 지방조직)**         |\n"],"metadata":{"id":"802wwC06_uRY"}},{"cell_type":"code","source":["expression_df=GSE77962_series_matrix_df"],"metadata":{"id":"un2zKsKtHRpC","executionInfo":{"status":"ok","timestamp":1750033353093,"user_tz":-540,"elapsed":9,"user":{"displayName":"권혜빈","userId":"01947391368488800076"}}},"execution_count":20,"outputs":[]},{"cell_type":"code","source":["expression_df.to_csv(\"/content/drive/MyDrive/BIO_MEDICAL_PROJECT/obesity_gene_analysis/data/expression_df.csv\")"],"metadata":{"id":"e1nnpeKlHdVo","executionInfo":{"status":"ok","timestamp":1750033404431,"user_tz":-540,"elapsed":9614,"user":{"displayName":"권혜빈","userId":"01947391368488800076"}}},"execution_count":21,"outputs":[]},{"cell_type":"markdown","source":["# Platform_df 데이터프레임"],"metadata":{"id":"gyVCfb2tDgRq"}},{"cell_type":"code","source":["# txt파일 처음 읽어보고 데이터 파악하기\n","platform_path = \"/content/drive/MyDrive/BIO_MEDICAL_PROJECT/obesity_gene_analysis/data/GPL11532-32230.txt\"\n","\n","with open(platform_path, 'r') as f:\n","    for i in range(40):\n","        line = f.readline()\n","        print(f\"{i}: {line.strip()}\")"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"9vN0fGS58Qh_","executionInfo":{"status":"ok","timestamp":1750030695151,"user_tz":-540,"elapsed":297,"user":{"displayName":"권혜빈","userId":"01947391368488800076"}},"outputId":"fcb59075-6175-47aa-9ecb-2e98afb7153e"},"execution_count":9,"outputs":[{"output_type":"stream","name":"stdout","text":["0: #ID = transcript_cluster_id\n","1: #GB_LIST = GenBank and RefSeq Accessions from mrna_assignment column\n","2: #SPOT_ID = genomic location of the transcript cluster in the version of the genome assembly used at annotation time. Coordinates are standard 1-based (length=stop-start+1).\n","3: #seqname = chromosome number\n","4: #RANGE_GB = NCBI RefSeq for chromosome of current build\n","5: #RANGE_STRAND = strand  (+|-)\n","6: #RANGE_START = start  (integer). Coordinates are standard 1-based (length=stop-start+1).\n","7: #RANGE_STOP = stop  (integer). Coordinates are standard 1-based (length=stop-start+1).\n","8: #total_probes = Total number of probes contained by this transcript cluster.\n","9: #gene_assignment = Gene information for each assigned mRNA for mRNAs that corresponds to known genes (multipart).\n","10: #mrna_assignment = Description of the public mRNAs that should be detected by the sets within this transcript cluster based on sequence alignment (multipart).\n","11: #category = Array design category of the transcript cluster\n","12: ID\tGB_LIST\tSPOT_ID\tseqname\tRANGE_GB\tRANGE_STRAND\tRANGE_START\tRANGE_STOP\ttotal_probes\tgene_assignment\tmrna_assignment\tcategory\n","13: 7896736\t\tchr1:53049-54936\tchr1\tNC_000001.10\t+\t53049\t54936\t7\t---\t---\t---\n","14: 7896738\t\tchr1:63015-63887\tchr1\tNC_000001.10\t+\t63015\t63887\t31\t---\tENST00000328113 // ENSEMBL // cdna:pseudogene chromosome:GRCh37:15:102467008:102467910:-1 gene:ENSG00000183909 // chr1 // 100 // 100 // 31 // 31 // 0 /// ENST00000318181 // ENSEMBL // cdna:pseudogene chromosome:GRCh37:19:104601:105256:1 gene:ENSG00000176705 // chr1 // 100 // 100 // 31 // 31 // 0 /// ENST00000492842 // ENSEMBL // cdna:pseudogene chromosome:GRCh37:1:62948:63887:1 gene:ENSG00000240361 // chr1 // 100 // 100 // 31 // 31 // 0\tmain\n","15: 7896740\tNM_001005240,NM_001004195,NM_001005484,BC136848,BC136907\tchr1:69091-70008\tchr1\tNC_000001.10\t+\t69091\t70008\t24\tNM_001005240 // OR4F17 // olfactory receptor, family 4, subfamily F, member 17 // 19p13.3 // 81099 /// NM_001004195 // OR4F4 // olfactory receptor, family 4, subfamily F, member 4 // 15q26.3 // 26682 /// NM_001005484 // OR4F5 // olfactory receptor, family 4, subfamily F, member 5 // 1p36.33 // 79501 /// ENST00000318050 // OR4F17 // olfactory receptor, family 4, subfamily F, member 17 // 19p13.3 // 81099 /// ENST00000335137 // OR4F4 // olfactory receptor, family 4, subfamily F, member 4 // 15q26.3 // 26682 /// ENST00000326183 // OR4F5 // olfactory receptor, family 4, subfamily F, member 5 // 1p36.33 // 79501 /// BC136848 // OR4F17 // olfactory receptor, family 4, subfamily F, member 17 // 19p13.3 // 81099 /// BC136907 // OR4F4 // olfactory receptor, family 4, subfamily F, member 4 // 15q26.3 // 26682 /// ENST00000442916 // OR4F17 // olfactory receptor, family 4, subfamily F, member 17 // 19p13.3 // 81099\tNM_001005240 // RefSeq // Homo sapiens olfactory receptor, family 4, subfamily F, member 17 (OR4F17), mRNA. // chr1 // 100 // 100 // 24 // 24 // 0 /// NM_001004195 // RefSeq // Homo sapiens olfactory receptor, family 4, subfamily F, member 4 (OR4F4), mRNA. // chr1 // 100 // 100 // 24 // 24 // 0 /// NM_001005484 // RefSeq // Homo sapiens olfactory receptor, family 4, subfamily F, member 5 (OR4F5), mRNA. // chr1 // 100 // 100 // 24 // 24 // 0 /// ENST00000318050 // ENSEMBL // Olfactory receptor 4F17 gene:ENSG00000176695 // chr1 // 100 // 100 // 24 // 24 // 0 /// ENST00000335137 // ENSEMBL // Olfactory receptor 4F4 gene:ENSG00000186092 // chr1 // 100 // 100 // 24 // 24 // 0 /// ENST00000326183 // ENSEMBL // Olfactory receptor 4F5 gene:ENSG00000177693 // chr1 // 100 // 100 // 24 // 24 // 0 /// BC136848 // GenBank // Homo sapiens olfactory receptor, family 4, subfamily F, member 17, mRNA (cDNA clone MGC:168462 IMAGE:9020839), complete cds. // chr1 // 100 // 100 // 24 // 24 // 0 /// BC136907 // GenBank // Homo sapiens olfactory receptor, family 4, subfamily F, member 4, mRNA (cDNA clone MGC:168521 IMAGE:9020898), complete cds. // chr1 // 100 // 100 // 24 // 24 // 0 /// ENST00000442916 // ENSEMBL // OR4F4 (Fragment) gene:ENSG00000176695 // chr1 // 100 // 88 // 21 // 21 // 0\tmain\n","16: 7896742\tBC118988,AL137655\tchr1:334129-334296\tchr1\tNC_000001.10\t+\t334129\t334296\t6\tENST00000388975 // SEPT14 // septin 14 // 7p11.2 // 346288 /// BC118988 // NCRNA00266 // non-protein coding RNA 266 // --- // 140849 /// AL137655 // LOC100134822 // similar to hCG1739109 // --- // 100134822\tENST00000388975 // ENSEMBL // Septin-14 gene:ENSG00000154997 // chr1 // 50 // 100 // 3 // 6 // 0 /// BC118988 // GenBank // Homo sapiens chromosome 20 open reading frame 69, mRNA (cDNA clone MGC:141807 IMAGE:40035995), complete cds. // chr1 // 100 // 100 // 6 // 6 // 0 /// AL137655 // GenBank // Homo sapiens mRNA; cDNA DKFZp434B2016 (from clone DKFZp434B2016). // chr1 // 100 // 100 // 6 // 6 // 0 /// ENST00000428915 // ENSEMBL // cdna:known chromosome:GRCh37:10:38742109:38755311:1 gene:ENSG00000203496 // chr1 // 100 // 100 // 6 // 6 // 0 /// ENST00000455207 // ENSEMBL // cdna:known chromosome:GRCh37:1:334129:446155:1 gene:ENSG00000224813 // chr1 // 100 // 100 // 6 // 6 // 0 /// ENST00000455464 // ENSEMBL // cdna:known chromosome:GRCh37:1:334140:342806:1 gene:ENSG00000224813 // chr1 // 100 // 100 // 6 // 6 // 0 /// ENST00000440200 // ENSEMBL // cdna:known chromosome:GRCh37:1:536816:655580:-1 gene:ENSG00000230021 // chr1 // 100 // 100 // 6 // 6 // 0 /// ENST00000279067 // ENSEMBL // cdna:known chromosome:GRCh37:20:62921738:62934912:1 gene:ENSG00000149656 // chr1 // 100 // 100 // 6 // 6 // 0 /// ENST00000499986 // ENSEMBL // cdna:known chromosome:GRCh37:5:180717576:180761371:1 gene:ENSG00000248628 // chr1 // 100 // 100 // 6 // 6 // 0 /// ENST00000436899 // ENSEMBL // cdna:known chromosome:GRCh37:6:131910:144885:-1 gene:ENSG00000170590 // chr1 // 100 // 100 // 6 // 6 // 0 /// ENST00000432557 // ENSEMBL // cdna:known chromosome:GRCh37:8:132324:150572:-1 gene:ENSG00000250210 // chr1 // 100 // 100 // 6 // 6 // 0 /// ENST00000523795 // ENSEMBL // cdna:known chromosome:GRCh37:8:141690:150563:-1 gene:ENSG00000250210 // chr1 // 100 // 100 // 6 // 6 // 0 /// ENST00000490482 // ENSEMBL // cdna:known chromosome:GRCh37:8:149942:163324:-1 gene:ENSG00000223508 // chr1 // 100 // 100 // 6 // 6 // 0 /// ENST00000307499 // ENSEMBL // cdna:known supercontig::GL000227.1:57780:70752:-1 gene:ENSG00000229450 // chr1 // 100 // 100 // 6 // 6 // 0 /// ENST00000441245 // ENSEMBL // cdna:known chromosome:GRCh37:1:637316:655530:-1 gene:ENSG00000230021 // chr1 // 100 // 67 // 4 // 4 // 0 /// ENST00000425473 // ENSEMBL // cdna:known chromosome:GRCh37:20:62926294:62944485:1 gene:ENSG00000149656 // chr1 // 100 // 67 // 4 // 4 // 0 /// ENST00000471248 // ENSEMBL // cdna:known chromosome:GRCh37:1:110953:129173:-1 gene:ENSG00000238009 // chr1 // 75 // 67 // 3 // 4 // 0\tmain\n","17: 7896744\tNM_001005277,NM_001005221,NM_001005224,NM_001005504,BC137547\tchr1:367659-368597\tchr1\tNC_000001.10\t+\t367659\t368597\t36\tNM_001005277 // OR4F16 // olfactory receptor, family 4, subfamily F, member 16 // 1p36.33 // 81399 /// NM_001005221 // OR4F29 // olfactory receptor, family 4, subfamily F, member 29 // 1p36.33 // 729759 /// NM_001005224 // OR4F3 // olfactory receptor, family 4, subfamily F, member 3 // 5q35.3 // 26683 /// NM_001005504 // OR4F21 // olfactory receptor, family 4, subfamily F, member 21 // 8p23.3 // 441308 /// ENST00000320901 // OR4F21 // olfactory receptor, family 4, subfamily F, member 21 // 8p23.3 // 441308 /// BC137547 // OR4F3 // olfactory receptor, family 4, subfamily F, member 3 // 5q35.3 // 26683 /// BC137547 // OR4F16 // olfactory receptor, family 4, subfamily F, member 16 // 1p36.33 // 81399 /// BC137547 // OR4F29 // olfactory receptor, family 4, subfamily F, member 29 // 1p36.33 // 729759\tNM_001005277 // RefSeq // Homo sapiens olfactory receptor, family 4, subfamily F, member 16 (OR4F16), mRNA. // chr1 // 100 // 100 // 36 // 36 // 0 /// NM_001005221 // RefSeq // Homo sapiens olfactory receptor, family 4, subfamily F, member 29 (OR4F29), mRNA. // chr1 // 100 // 100 // 36 // 36 // 0 /// NM_001005224 // RefSeq // Homo sapiens olfactory receptor, family 4, subfamily F, member 3 (OR4F3), mRNA. // chr1 // 100 // 100 // 36 // 36 // 0 /// NM_001005504 // RefSeq // Homo sapiens olfactory receptor, family 4, subfamily F, member 21 (OR4F21), mRNA. // chr1 // 89 // 100 // 32 // 36 // 0 /// ENST00000320901 // ENSEMBL // Olfactory receptor 4F21 gene:ENSG00000176269 // chr1 // 89 // 100 // 32 // 36 // 0 /// BC137547 // GenBank // Homo sapiens olfactory receptor, family 4, subfamily F, member 3, mRNA (cDNA clone MGC:169170 IMAGE:9021547), complete cds. // chr1 // 100 // 100 // 36 // 36 // 0 /// ENST00000426406 // ENSEMBL // cdna:known chromosome:GRCh37:1:367640:368634:1 gene:ENSG00000235249 // chr1 // 100 // 100 // 36 // 36 // 0 /// ENST00000332831 // ENSEMBL // cdna:known chromosome:GRCh37:1:621096:622034:-1 gene:ENSG00000185097 // chr1 // 100 // 100 // 36 // 36 // 0 /// ENST00000456475 // ENSEMBL // cdna:known chromosome:GRCh37:5:180794269:180795263:1 gene:ENSG00000230178 // chr1 // 100 // 100 // 36 // 36 // 0 /// ENST00000521196 // ENSEMBL // cdna:known chromosome:GRCh37:11:86612:87605:-1 gene:ENSG00000224777 // chr1 // 78 // 100 // 28 // 36 // 0\tmain\n","18: 7896746\tNC_001807,AK074482\tchr1:564951-565019\tchr1\tNC_000001.10\t+\t564951\t565019\t28\t---\tNC_001807 // GenBank // gi|17981852|ref|NC_001807.4|:4403-4470; gene=TRNM; product=tRNA-Met // chr1 // 78 // 96 // 21 // 27 // 0 /// AK074482 // GenBank // Homo sapiens cDNA fis, A-BNGH41000020, highly similar to Homo sapiens isolate EIJbIII-4-94 NADH dehydrogenase subunit 1 gene; mitochondrial gene for mitochondrial product. // chr1 // 79 // 100 // 22 // 28 // 0 /// ENST00000467964 // ENSEMBL // ncrna_pseudogene:Mt_tRNA_pseudogene chromosome:GRCh37:17:22025157:22025224:1 gene:ENSG00000240259 // chr1 // 26 // 96 // 7 // 27 // 1 /// ENST00000468070 // ENSEMBL // ncrna_pseudogene:Mt_tRNA_pseudogene chromosome:GRCh37:2:140977938:140978003:-1 gene:ENSG00000240717 // chr1 // 28 // 64 // 5 // 18 // 1\tmain\n","19: 7896748\tNC_001807\tchr1:566062-566129\tchr1\tNC_000001.10\t+\t566062\t566129\t24\t---\tNC_001807 // GenBank // gi|17981852|ref|NC_001807.4|:5513-5580; gene=TRNW; product=tRNA-Trp // chr1 // 100 // 100 // 24 // 24 // 0\tmain\n","20: 7896750\tNC_001807,AK172782\tchr1:568069-568136\tchr1\tNC_000001.10\t+\t568069\t568136\t28\tAK172782 // GPAM // glycerol-3-phosphate acyltransferase, mitochondrial // 10q25.2 // 57678\tNC_001807 // GenBank // gi|17981852|ref|NC_001807.4|:7519-7586; gene=TRND; product=tRNA-Asp // chr1 // 86 // 100 // 24 // 28 // 0 /// AK172782 // GenBank // Homo sapiens cDNA FLJ23943 fis, clone HEP04035, highly similar to Glycerol-3-phosphate acyltransferase, mitochondrial precursor (EC 2.3.1.15). // chr1 // 100 // 25 // 7 // 7 // 0\tmain\n","21: 7896752\tNC_001807,M37726\tchr1:568844-568913\tchr1\tNC_000001.10\t+\t568844\t568913\t24\t---\tNC_001807 // GenBank // gi|17981852|ref|NC_001807.4|:8296-8365; gene=TRNK; product=tRNA-Lys // chr1 // 100 // 100 // 24 // 24 // 0 /// M37726 // GenBank // Human mitochondrial Lys-tRNA-aaa. // chr1 // 100 // 100 // 24 // 24 // 0\tmain\n","22: 7896754\tAK290103,AK026836,AK124970\tchr1:721406-721912\tchr1\tNC_000001.10\t+\t721406\t721912\t23\tAK290103 // LOC100287934 // hypothetical LOC100287934 // 1p36.33 // 100287934 /// ENST00000358533 // LOC100287934 // hypothetical LOC100287934 // 1p36.33 // 100287934\tAK290103 // GenBank // Homo sapiens cDNA FLJ78588 complete cds. // chr1 // 87 // 100 // 20 // 23 // 0 /// ENST00000358533 // ENSEMBL // Transmembrane protein FLJ78588 gene:ENSG00000197049 // chr1 // 100 // 100 // 23 // 23 // 0 /// AK026836 // GenBank // Homo sapiens cDNA: FLJ23183 fis, clone LNG11477. // chr1 // 100 // 100 // 23 // 23 // 0 /// AK124970 // GenBank // Homo sapiens cDNA FLJ42980 fis, clone BRTHA2006735. // chr1 // 100 // 17 // 4 // 4 // 0 /// ENST00000449219 // ENSEMBL // cdna:known chromosome:GRCh37:1:224183659:224197405:1 gene:ENSG00000185495 // chr1 // 100 // 17 // 4 // 4 // 0\tmain\n","23: 7896756\tBC037297,BC118644\tchr1:752751-755214\tchr1\tNC_000001.10\t+\t752751\t755214\t26\tBC037297 // FAM87A // family with sequence similarity 87, member A // 8p23.3 // 157693\tBC037297 // GenBank // Homo sapiens cDNA clone IMAGE:4821387. // chr1 // 88 // 100 // 23 // 26 // 0 /// ENST00000330148 // ENSEMBL // hypothetical protein gene:ENSG00000182366 // chr1 // 88 // 100 // 23 // 26 // 0 /// ENST00000523195 // ENSEMBL // cdna:known chromosome:GRCh37:8:326043:333174:-1 gene:ENSG00000182366 // chr1 // 88 // 96 // 22 // 25 // 0 /// BC118644 // GenBank // Homo sapiens cDNA clone IMAGE:40030978. // chr1 // 90 // 38 // 9 // 10 // 0\tmain\n","24: 7896759\tAK096570\tchr1:791391-793751\tchr1\tNC_000001.10\t+\t791391\t793751\t12\tAK096570 // LOC643837 // hypothetical LOC643837 // 1p36.33 // 643837\tAK096570 // GenBank // Homo sapiens cDNA FLJ39251 fis, clone OCBBF2008701. // chr1 // 100 // 100 // 12 // 12 // 0\tmain\n","25: 7896761\tNM_152486,NM_015658,BC024295\tchr1:860530-880512\tchr1\tNC_000001.10\t+\t860530\t880512\t42\tNM_152486 // SAMD11 // sterile alpha motif domain containing 11 // 1p36.33 // 148398 /// NM_015658 // NOC2L // nucleolar complex associated 2 homolog (S. cerevisiae) // 1p36.33 // 26155 /// ENST00000342066 // SAMD11 // sterile alpha motif domain containing 11 // 1p36.33 // 148398 /// ENST00000327044 // NOC2L // nucleolar complex associated 2 homolog (S. cerevisiae) // 1p36.33 // 26155 /// BC024295 // SAMD11 // sterile alpha motif domain containing 11 // 1p36.33 // 148398 /// ENST00000443100 // SAMD11 // sterile alpha motif domain containing 11 // 1p36.33 // 148398 /// ENST00000420190 // SAMD11 // sterile alpha motif domain containing 11 // 1p36.33 // 148398 /// ENST00000437963 // SAMD11 // sterile alpha motif domain containing 11 // 1p36.33 // 148398\tNM_152486 // RefSeq // Homo sapiens sterile alpha motif domain containing 11 (SAMD11), mRNA. // chr1 // 100 // 100 // 42 // 42 // 0 /// NM_015658 // RefSeq // Homo sapiens nucleolar complex associated 2 homolog (S. cerevisiae) (NOC2L), mRNA. // chr1 // 100 // 2 // 2 // 1 // 0 /// ENST00000342066 // ENSEMBL // Isoform 3 of Sterile alpha motif domain-containing protein 11 gene:ENSG00000187634 // chr1 // 100 // 100 // 42 // 42 // 0 /// ENST00000327044 // ENSEMBL // Nucleolar complex protein 2 homolog gene:ENSG00000188976 // chr1 // 100 // 2 // 2 // 1 // 0 /// BC024295 // GenBank // Homo sapiens sterile alpha motif domain containing 11, mRNA (cDNA clone MGC:39333 IMAGE:3354502), complete cds. // chr1 // 100 // 71 // 30 // 30 // 0 /// ENST00000443100 // ENSEMBL // Isoform 2 of Sterile alpha motif domain-containing protein 11 gene:ENSG00000187634 // chr1 // 100 // 90 // 38 // 38 // 0 /// ENST00000420190 // ENSEMBL // Sterile alpha motif domain containing 11 gene:ENSG00000187634 // chr1 // 100 // 24 // 10 // 10 // 0 /// ENST00000437963 // ENSEMBL // Sterile alpha motif domain containing 11 gene:ENSG00000187634 // chr1 // 100 // 14 // 6 // 6 // 0 /// ENST00000478729 // ENSEMBL // cdna:known chromosome:GRCh37:1:875726:877553:1 gene:ENSG00000187634 // chr1 // 100 // 12 // 5 // 5 // 0 /// ENST00000466827 // ENSEMBL // cdna:known chromosome:GRCh37:1:877483:878182:1 gene:ENSG00000187634 // chr1 // 100 // 10 // 4 // 4 // 0 /// ENST00000464948 // ENSEMBL // cdna:known chromosome:GRCh37:1:877546:878272:1 gene:ENSG00000187634 // chr1 // 100 // 10 // 4 // 4 // 0\tmain\n","26: 7896779\tNM_198317,AY423763\tchr1:895967-901099\tchr1\tNC_000001.10\t+\t895967\t901099\t47\tNM_198317 // KLHL17 // kelch-like 17 (Drosophila) // 1p36.33 // 339451 /// ENST00000338591 // KLHL17 // kelch-like 17 (Drosophila) // 1p36.33 // 339451 /// AY423763 // KLHL17 // kelch-like 17 (Drosophila) // 1p36.33 // 339451 /// ENST00000455747 // KLHL17 // kelch-like 17 (Drosophila) // 1p36.33 // 339451\tNM_198317 // RefSeq // Homo sapiens kelch-like 17 (Drosophila) (KLHL17), mRNA. // chr1 // 100 // 83 // 39 // 39 // 0 /// ENST00000338591 // ENSEMBL // Kelch-like protein 17 gene:ENSG00000187961 // chr1 // 100 // 83 // 39 // 39 // 0 /// AY423763 // GenBank // Homo sapiens actinfilin mRNA, complete cds. // chr1 // 100 // 83 // 39 // 39 // 0 /// ENST00000455747 // ENSEMBL // KLHL17 protein gene:ENSG00000187961 // chr1 // 100 // 30 // 14 // 14 // 0 /// ENST00000473277 // ENSEMBL // cdna:known chromosome:GRCh37:1:895967:901095:1 gene:ENSG00000187961 // chr1 // 100 // 70 // 33 // 33 // 0 /// ENST00000463212 // ENSEMBL // cdna:known chromosome:GRCh37:1:896829:897858:1 gene:ENSG00000187961 // chr1 // 100 // 32 // 15 // 15 // 0 /// ENST00000466300 // ENSEMBL // cdna:known chromosome:GRCh37:1:898107:899910:1 gene:ENSG00000187961 // chr1 // 100 // 28 // 13 // 13 // 0\tmain\n","27: 7896798\tNM_032129,NM_001160184,BC101387\tchr1:901877-910488\tchr1\tNC_000001.10\t+\t901877\t910488\t43\tNM_032129 // PLEKHN1 // pleckstrin homology domain containing, family N member 1 // 1p36.33 // 84069 /// NM_001160184 // PLEKHN1 // pleckstrin homology domain containing, family N member 1 // 1p36.33 // 84069 /// ENST00000379410 // PLEKHN1 // pleckstrin homology domain containing, family N member 1 // 1p36.33 // 84069 /// ENST00000379407 // PLEKHN1 // pleckstrin homology domain containing, family N member 1 // 1p36.33 // 84069 /// BC101387 // PLEKHN1 // pleckstrin homology domain containing, family N member 1 // 1p36.33 // 84069 /// ENST00000379409 // PLEKHN1 // pleckstrin homology domain containing, family N member 1 // 1p36.33 // 84069\tNM_032129 // RefSeq // Homo sapiens pleckstrin homology domain containing, family N member 1 (PLEKHN1), transcript variant 1, mRNA. // chr1 // 100 // 91 // 39 // 39 // 0 /// NM_001160184 // RefSeq // Homo sapiens pleckstrin homology domain containing, family N member 1 (PLEKHN1), transcript variant 2, mRNA. // chr1 // 100 // 91 // 39 // 39 // 0 /// ENST00000379410 // ENSEMBL // Isoform 2 of Pleckstrin homology domain-containing family N member 1 gene:ENSG00000187583 // chr1 // 100 // 91 // 39 // 39 // 0 /// ENST00000379407 // ENSEMBL // Isoform 3 of Pleckstrin homology domain-containing family N member 1 gene:ENSG00000187583 // chr1 // 100 // 91 // 39 // 39 // 0 /// BC101387 // GenBank // Homo sapiens pleckstrin homology domain containing, family N member 1, mRNA (cDNA clone MGC:120616 IMAGE:40026404), complete cds. // chr1 // 95 // 95 // 39 // 41 // 0 /// ENST00000379409 // ENSEMBL // Isoform 1 of Pleckstrin homology domain-containing family N member 1 gene:ENSG00000187583 // chr1 // 100 // 100 // 43 // 43 // 0 /// ENST00000480267 // ENSEMBL // cdna:known chromosome:GRCh37:1:906255:906903:1 gene:ENSG00000187583 // chr1 // 100 // 23 // 10 // 10 // 0 /// ENST00000491024 // ENSEMBL // cdna:known chromosome:GRCh37:1:908892:911245:1 gene:ENSG00000187583 // chr1 // 100 // 19 // 8 // 8 // 0\tmain\n","28: 7896817\tNM_005101,M13755\tchr1:948803-949920\tchr1\tNC_000001.10\t+\t948803\t949920\t27\tNM_005101 // ISG15 // ISG15 ubiquitin-like modifier // 1p36.33 // 9636 /// ENST00000379389 // ISG15 // ISG15 ubiquitin-like modifier // 1p36.33 // 9636 /// M13755 // ISG15 // ISG15 ubiquitin-like modifier // 1p36.33 // 9636\tNM_005101 // RefSeq // Homo sapiens ISG15 ubiquitin-like modifier (ISG15), mRNA. // chr1 // 100 // 100 // 27 // 27 // 0 /// ENST00000379389 // ENSEMBL // Ubiquitin-like protein ISG15 gene:ENSG00000187608 // chr1 // 100 // 100 // 27 // 27 // 0 /// M13755 // GenBank // Human interferon-induced 17-kDa/15-kDa protein mRNA, complete cds. // chr1 // 88 // 96 // 23 // 26 // 0\tmain\n","29: 7896822\tNM_198576,AB191264\tchr1:955503-991496\tchr1\tNC_000001.10\t+\t955503\t991496\t49\tNM_198576 // AGRN // agrin // 1p36.33 // 375790 /// ENST00000379370 // AGRN // agrin // 1p36.33 // 375790 /// AB191264 // AGRN // agrin // 1p36.33 // 375790 /// ENST00000379364 // AGRN // agrin // 1p36.33 // 375790 /// ENST00000419249 // AGRN // agrin // 1p36.33 // 375790\tNM_198576 // RefSeq // Homo sapiens agrin (AGRN), mRNA. // chr1 // 100 // 100 // 49 // 49 // 0 /// ENST00000379370 // ENSEMBL // Agrin gene:ENSG00000188157 // chr1 // 100 // 100 // 49 // 49 // 0 /// AB191264 // GenBank // Homo sapiens AGRN mRNA for agrin, complete cds. // chr1 // 98 // 100 // 48 // 49 // 0 /// ENST00000379364 // ENSEMBL // Agrin gene:ENSG00000188157 // chr1 // 100 // 100 // 49 // 49 // 0 /// ENST00000419249 // ENSEMBL // Agrin gene:ENSG00000188157 // chr1 // 100 // 12 // 6 // 6 // 0 /// ENST00000461111 // ENSEMBL // cdna:known chromosome:GRCh37:1:987366:991492:1 gene:ENSG00000188157 // chr1 // 100 // 20 // 10 // 10 // 0 /// ENST00000479707 // ENSEMBL // cdna:known chromosome:GRCh37:1:980779:981729:1 gene:ENSG00000188157 // chr1 // 100 // 10 // 5 // 5 // 0 /// ENST00000469403 // ENSEMBL // cdna:known chromosome:GRCh37:1:970247:976777:1 gene:ENSG00000188157 // chr1 // 100 // 6 // 3 // 3 // 0 /// ENST00000466223 // ENSEMBL // cdna:known chromosome:GRCh37:1:982581:983245:1 gene:ENSG00000188157 // chr1 // 100 // 6 // 3 // 3 // 0\tmain\n","30: 7896859\tNR_029639\tchr1:1102484-1102578\tchr1\tNC_000001.10\t+\t1102484\t1102578\t25\tNR_029639 // MIR200B // microRNA 200b // 1p36.33 // 406984\tNR_029639 // RefSeq // Homo sapiens microRNA 200b (MIR200B), microRNA. // chr1 // 100 // 100 // 25 // 25 // 0 /// hsa-mir-200b // miRBase Micro RNA Database // MI0000342 Homo sapiens miR-200b stem-loop // chr1 // 100 // 100 // 25 // 25 // 0\tmain\n","31: 7896861\tNR_029834\tchr1:1103243-1103332\tchr1\tNC_000001.10\t+\t1103243\t1103332\t25\tNR_029834 // MIR200A // microRNA 200a // 1p36.33 // 406983\tNR_029834 // RefSeq // Homo sapiens microRNA 200a (MIR200A), microRNA. // chr1 // 100 // 100 // 25 // 25 // 0 /// hsa-mir-200a // miRBase Micro RNA Database // MI0000737 Homo sapiens miR-200a stem-loop // chr1 // 100 // 100 // 25 // 25 // 0\tmain\n","32: 7896863\tNR_029957\tchr1:1104373-1104471\tchr1\tNC_000001.10\t+\t1104373\t1104471\t18\tNR_029957 // MIR429 // microRNA 429 // 1p36.33 // 554210\tNR_029957 // RefSeq // Homo sapiens microRNA 429 (MIR429), microRNA. // chr1 // 100 // 89 // 16 // 16 // 0 /// hsa-mir-429 // miRBase Micro RNA Database // MI0001641 Homo sapiens miR-429 stem-loop // chr1 // 100 // 89 // 16 // 16 // 0\tmain\n","33: 7896865\tNM_153254,NM_001130045,BC126152\tchr1:1109283-1133315\tchr1\tNC_000001.10\t+\t1109283\t1133315\t32\tNM_153254 // TTLL10 // tubulin tyrosine ligase-like family, member 10 // 1p36.33 // 254173 /// NM_001130045 // TTLL10 // tubulin tyrosine ligase-like family, member 10 // 1p36.33 // 254173 /// ENST00000379288 // TTLL10 // tubulin tyrosine ligase-like family, member 10 // 1p36.33 // 254173 /// ENST00000400931 // TTLL10 // tubulin tyrosine ligase-like family, member 10 // 1p36.33 // 254173 /// ENST00000379290 // TTLL10 // tubulin tyrosine ligase-like family, member 10 // 1p36.33 // 254173 /// ENST00000379289 // TTLL10 // tubulin tyrosine ligase-like family, member 10 // 1p36.33 // 254173 /// BC126152 // TTLL10 // tubulin tyrosine ligase-like family, member 10 // 1p36.33 // 254173\tNM_153254 // RefSeq // Homo sapiens tubulin tyrosine ligase-like family, member 10 (TTLL10), transcript variant 2, mRNA. // chr1 // 100 // 94 // 30 // 30 // 0 /// NM_001130045 // RefSeq // Homo sapiens tubulin tyrosine ligase-like family, member 10 (TTLL10), transcript variant 1, mRNA. // chr1 // 100 // 56 // 18 // 18 // 0 /// ENST00000379288 // ENSEMBL // Isoform 3 of Inactive polyglycylase TTLL10 gene:ENSG00000162571 // chr1 // 100 // 94 // 30 // 30 // 0 /// ENST00000400931 // ENSEMBL // Isoform 3 of Inactive polyglycylase TTLL10 gene:ENSG00000162571 // chr1 // 100 // 63 // 20 // 20 // 0 /// ENST00000379290 // ENSEMBL // Isoform 1 of Inactive polyglycylase TTLL10 gene:ENSG00000162571 // chr1 // 100 // 56 // 18 // 18 // 0 /// ENST00000379289 // ENSEMBL // Isoform 1 of Inactive polyglycylase TTLL10 gene:ENSG00000162571 // chr1 // 100 // 56 // 18 // 18 // 0 /// BC126152 // GenBank // Homo sapiens tubulin tyrosine ligase-like family, member 10, mRNA (cDNA clone MGC:161430 IMAGE:8991868), complete cds. // chr1 // 100 // 91 // 29 // 29 // 0 /// ENST00000460998 // ENSEMBL // cdna:known chromosome:GRCh37:1:1115864:1119307:1 gene:ENSG00000162571 // chr1 // 100 // 31 // 10 // 10 // 0\tmain\n","34: 7896878\tNM_080605,BC015549\tchr1:1167620-1170421\tchr1\tNC_000001.10\t+\t1167620\t1170421\t27\tNM_080605 // B3GALT6 // UDP-Gal:betaGal beta 1,3-galactosyltransferase polypeptide 6 // 1p36.33 // 126792 /// ENST00000379198 // B3GALT6 // UDP-Gal:betaGal beta 1,3-galactosyltransferase polypeptide 6 // 1p36.33 // 126792 /// BC015549 // B3GALT6 // UDP-Gal:betaGal beta 1,3-galactosyltransferase polypeptide 6 // 1p36.33 // 126792\tNM_080605 // RefSeq // Homo sapiens UDP-Gal:betaGal beta 1,3-galactosyltransferase polypeptide 6 (B3GALT6), mRNA. // chr1 // 100 // 89 // 24 // 24 // 0 /// ENST00000379198 // ENSEMBL // Beta-1,3-galactosyltransferase 6 gene:ENSG00000176022 // chr1 // 100 // 89 // 24 // 24 // 0 /// BC015549 // GenBank // Homo sapiens UDP-Gal:betaGal beta 1,3-galactosyltransferase polypeptide 6, mRNA (cDNA clone IMAGE:3846772). // chr1 // 100 // 22 // 6 // 6 // 0\tmain\n","35: 7896882\tNM_001130413,NM_002978,AK127357\tchr1:1215968-1227409\tchr1\tNC_000001.10\t+\t1215968\t1227409\t36\tNM_001130413 // SCNN1D // sodium channel, nonvoltage-gated 1, delta // 1p36.3-p36.2 // 6339 /// NM_002978 // SCNN1D // sodium channel, nonvoltage-gated 1, delta // 1p36.3-p36.2 // 6339 /// ENST00000338555 // SCNN1D // sodium channel, nonvoltage-gated 1, delta // 1p36.3-p36.2 // 6339 /// ENST00000325425 // SCNN1D // sodium channel, nonvoltage-gated 1, delta // 1p36.3-p36.2 // 6339 /// ENST00000400928 // SCNN1D // sodium channel, nonvoltage-gated 1, delta // 1p36.3-p36.2 // 6339 /// AK127357 // SCNN1D // sodium channel, nonvoltage-gated 1, delta // 1p36.3-p36.2 // 6339 /// ENST00000379110 // SCNN1D // sodium channel, nonvoltage-gated 1, delta // 1p36.3-p36.2 // 6339 /// ENST00000379116 // SCNN1D // sodium channel, nonvoltage-gated 1, delta // 1p36.3-p36.2 // 6339 /// ENST00000379099 // SCNN1D // sodium channel, nonvoltage-gated 1, delta // 1p36.3-p36.2 // 6339\tNM_001130413 // RefSeq // Homo sapiens sodium channel, nonvoltage-gated 1, delta (SCNN1D), transcript variant 1, mRNA. // chr1 // 100 // 81 // 29 // 29 // 0 /// NM_002978 // RefSeq // Homo sapiens sodium channel, nonvoltage-gated 1, delta (SCNN1D), transcript variant 2, mRNA. // chr1 // 100 // 69 // 25 // 25 // 0 /// ENST00000338555 // ENSEMBL // Isoform 1 of Amiloride-sensitive sodium channel subunit delta gene:ENSG00000162572 // chr1 // 100 // 86 // 31 // 31 // 0 /// ENST00000325425 // ENSEMBL // Isoform 2 of Amiloride-sensitive sodium channel subunit delta gene:ENSG00000162572 // chr1 // 100 // 81 // 29 // 29 // 0 /// ENST00000400928 // ENSEMBL // Isoform 1 of Amiloride-sensitive sodium channel subunit delta gene:ENSG00000162572 // chr1 // 100 // 75 // 27 // 27 // 0 /// AK127357 // GenBank // Homo sapiens cDNA FLJ45431 fis, clone BRHIP3039509, highly similar to Amiloride-sensitive sodium channel subunit delta. // chr1 // 100 // 83 // 30 // 30 // 0 /// ENST00000379110 // ENSEMBL // cDNA FLJ16802 fis, clone TESTI4005158, highly similar to Amiloride-sensitive sodium channel delta-subunit gene:ENSG00000162572 // chr1 // 100 // 53 // 19 // 19 // 0 /// ENST00000379116 // ENSEMBL // Uncharacterized protein gene:ENSG00000162572 // chr1 // 100 // 39 // 14 // 14 // 0 /// ENST00000379099 // ENSEMBL // Sodium channel, nonvoltage-gated 1, delta gene:ENSG00000162572 // chr1 // 100 // 25 // 9 // 9 // 0 /// ENST00000379101 // ENSEMBL // cdna:known chromosome:GRCh37:1:1215852:1226753:1 gene:ENSG00000162572 // chr1 // 100 // 61 // 22 // 22 // 0 /// ENST00000467651 // ENSEMBL // cdna:known chromosome:GRCh37:1:1217544:1220981:1 gene:ENSG00000162572 // chr1 // 100 // 11 // 4 // 4 // 0\tmain\n","36: 7896908\tNM_153339,NM_017871,BC034304\tchr1:1243947-1247057\tchr1\tNC_000001.10\t+\t1243947\t1247057\t26\tNM_153339 // PUSL1 // pseudouridylate synthase-like 1 // 1p36.33 // 126789 /// NM_017871 // CPSF3L // cleavage and polyadenylation specific factor 3-like // 1p36.33 // 54973 /// ENST00000379031 // PUSL1 // pseudouridylate synthase-like 1 // 1p36.33 // 126789 /// ENST00000435064 // CPSF3L // cleavage and polyadenylation specific factor 3-like // 1p36.33 // 54973 /// BC034304 // PUSL1 // pseudouridylate synthase-like 1 // 1p36.33 // 126789\tNM_153339 // RefSeq // Homo sapiens pseudouridylate synthase-like 1 (PUSL1), mRNA. // chr1 // 100 // 100 // 26 // 26 // 0 /// NM_017871 // RefSeq // Homo sapiens cleavage and polyadenylation specific factor 3-like (CPSF3L), mRNA. // chr1 // 100 // 8 // 2 // 2 // 0 /// ENST00000379031 // ENSEMBL // Isoform 1 of tRNA pseudouridine synthase-like 1 gene:ENSG00000169972 // chr1 // 100 // 100 // 26 // 26 // 0 /// ENST00000435064 // ENSEMBL // Isoform 1 of Integrator complex subunit 11 gene:ENSG00000127054 // chr1 // 100 // 8 // 2 // 2 // 0 /// BC034304 // GenBank // Homo sapiens pseudouridylate synthase-like 1, mRNA (cDNA clone MGC:30098 IMAGE:4778177), complete cds. // chr1 // 100 // 100 // 26 // 26 // 0 /// ENST00000493657 // ENSEMBL // cdna:known chromosome:GRCh37:1:1244615:1247056:1 gene:ENSG00000169972 // chr1 // 100 // 62 // 16 // 16 // 0 /// ENST00000470520 // ENSEMBL // cdna:known chromosome:GRCh37:1:1243991:1246408:1 gene:ENSG00000169972 // chr1 // 100 // 58 // 15 // 15 // 0 /// ENST00000467712 // ENSEMBL // cdna:known chromosome:GRCh37:1:1244201:1246449:1 gene:ENSG00000169972 // chr1 // 100 // 54 // 14 // 14 // 0\tmain\n","37: 7896917\tNM_001029885\tchr1:1260143-1264277\tchr1\tNC_000001.10\t+\t1260143\t1264277\t46\tNM_001029885 // GLTPD1 // glycolipid transfer protein domain containing 1 // 1p36.33 // 80772 /// ENST00000343938 // GLTPD1 // glycolipid transfer protein domain containing 1 // 1p36.33 // 80772\tNM_001029885 // RefSeq // Homo sapiens glycolipid transfer protein domain containing 1 (GLTPD1), mRNA. // chr1 // 100 // 100 // 46 // 46 // 0 /// ENST00000343938 // ENSEMBL // Glycolipid transfer protein domain-containing protein 1 gene:ENSG00000224051 // chr1 // 100 // 100 // 46 // 46 // 0 /// ENST00000464957 // ENSEMBL // cdna:known chromosome:GRCh37:1:1260136:1263374:1 gene:ENSG00000224051 // chr1 // 100 // 57 // 26 // 26 // 0 /// ENST00000488011 // ENSEMBL // cdna:known chromosome:GRCh37:1:1260621:1262851:1 gene:ENSG00000224051 // chr1 // 100 // 37 // 17 // 17 // 0\tmain\n","38: 7896921\tNM_152228,AK122959\tchr1:1266694-1270687\tchr1\tNC_000001.10\t+\t1266694\t1270687\t40\tNM_152228 // TAS1R3 // taste receptor, type 1, member 3 // 1p36.33 // 83756 /// ENST00000339381 // TAS1R3 // taste receptor, type 1, member 3 // 1p36.33 // 83756 /// AK122959 // TAS1R3 // taste receptor, type 1, member 3 // 1p36.33 // 83756\tNM_152228 // RefSeq // Homo sapiens taste receptor, type 1, member 3 (TAS1R3), mRNA. // chr1 // 100 // 85 // 34 // 34 // 0 /// ENST00000339381 // ENSEMBL // Taste receptor type 1 member 3 gene:ENSG00000169962 // chr1 // 100 // 100 // 40 // 40 // 0 /// AK122959 // GenBank // Homo sapiens cDNA FLJ16679 fis, clone TLIVE2003381, moderately similar to Mus musculus taste receptor, type 1, member 3 (Tas1r3). // chr1 // 89 // 45 // 16 // 18 // 0\tmain\n","39: 7896929\tNM_022834,NM_199121,BC059409\tchr1:1370909-1378262\tchr1\tNC_000001.10\t+\t1370909\t1378262\t41\tNM_022834 // VWA1 // von Willebrand factor A domain containing 1 // 1p36.33 // 64856 /// NM_199121 // VWA1 // von Willebrand factor A domain containing 1 // 1p36.33 // 64856 /// ENST00000476993 // VWA1 // von Willebrand factor A domain containing 1 // 1p36.33 // 64856 /// BC059409 // VWA1 // von Willebrand factor A domain containing 1 // 1p36.33 // 64856 /// ENST00000339380 // VWA1 // von Willebrand factor A domain containing 1 // 1p36.33 // 64856 /// ENST00000404702 // VWA1 // von Willebrand factor A domain containing 1 // 1p36.33 // 64856\tNM_022834 // RefSeq // Homo sapiens von Willebrand factor A domain containing 1 (VWA1), transcript variant 1, mRNA. // chr1 // 100 // 100 // 41 // 41 // 0 /// NM_199121 // RefSeq // Homo sapiens von Willebrand factor A domain containing 1 (VWA1), transcript variant 2, mRNA. // chr1 // 100 // 90 // 37 // 37 // 0 /// ENST00000476993 // ENSEMBL // Isoform 1 of von Willebrand factor A domain-containing protein 1 gene:ENSG00000179403 // chr1 // 100 // 93 // 38 // 38 // 0 /// BC059409 // GenBank // Homo sapiens von Willebrand factor A domain containing 1, mRNA (cDNA clone MGC:71828 IMAGE:30349211), complete cds. // chr1 // 100 // 85 // 35 // 35 // 0 /// ENST00000339380 // ENSEMBL // Isoform 2 of von Willebrand factor A domain-containing protein 1 gene:ENSG00000179403 // chr1 // 100 // 85 // 35 // 35 // 0 /// ENST00000404702 // ENSEMBL // Isoform 2 of von Willebrand factor A domain-containing protein 1 gene:ENSG00000179403 // chr1 // 100 // 83 // 34 // 34 // 0\tmain\n"]}]},{"cell_type":"code","source":["# 0~12행까지 주석이었으니 skiprows=12을 통해 정확한 헤더 위치를 지정\n","GPL11532_32230_df = pd.read_csv(platform_path, sep=\"\\t\", skiprows=12, low_memory=False)"],"metadata":{"id":"X6fWUKw09RnL","executionInfo":{"status":"ok","timestamp":1750032092319,"user_tz":-540,"elapsed":1182,"user":{"displayName":"권혜빈","userId":"01947391368488800076"}}},"execution_count":15,"outputs":[]},{"cell_type":"code","source":["GPL11532_32230_df"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":825},"id":"i2OZpYn293HK","executionInfo":{"status":"ok","timestamp":1750032093745,"user_tz":-540,"elapsed":548,"user":{"displayName":"권혜빈","userId":"01947391368488800076"}},"outputId":"4c3a8821-ef9a-4a3b-e874-0b83fb9ed675"},"execution_count":16,"outputs":[{"output_type":"execute_result","data":{"text/plain":["            ID                                            GB_LIST  \\\n","0      7896736                                                NaN   \n","1      7896738                                                NaN   \n","2      7896740  NM_001005240,NM_001004195,NM_001005484,BC13684...   \n","3      7896742                                  BC118988,AL137655   \n","4      7896744  NM_001005277,NM_001005221,NM_001005224,NM_0010...   \n","...        ...                                                ...   \n","33292  8180416                                        NM_002551.3   \n","33293  8180417                                        NM_018396.2   \n","33294  8180418                                     NM_001040145.1   \n","33295  7976552                                                NaN   \n","33296  8020352                                                NaN   \n","\n","                       SPOT_ID seqname      RANGE_GB RANGE_STRAND  \\\n","0             chr1:53049-54936    chr1  NC_000001.10            +   \n","1             chr1:63015-63887    chr1  NC_000001.10            +   \n","2             chr1:69091-70008    chr1  NC_000001.10            +   \n","3           chr1:334129-334296    chr1  NC_000001.10            +   \n","4           chr1:367659-368597    chr1  NC_000001.10            +   \n","...                        ...     ...           ...          ...   \n","33292         flmrna->unmapped     ---           NaN          ---   \n","33293         flmrna->unmapped     ---           NaN          ---   \n","33294         flmrna->unmapped     ---           NaN          ---   \n","33295  chr14:96154516-96154596   chr14   NC_000014.8            +   \n","33296  chr18:14341593-14342513   chr18   NC_000018.9            +   \n","\n","       RANGE_START  RANGE_STOP  total_probes  \\\n","0            53049       54936             7   \n","1            63015       63887            31   \n","2            69091       70008            24   \n","3           334129      334296             6   \n","4           367659      368597            36   \n","...            ...         ...           ...   \n","33292            0           0            39   \n","33293            0           0            33   \n","33294            0           0             6   \n","33295     96154516    96154596             4   \n","33296     14341593    14342513             9   \n","\n","                                         gene_assignment  \\\n","0                                                    ---   \n","1                                                    ---   \n","2      NM_001005240 // OR4F17 // olfactory receptor, ...   \n","3      ENST00000388975 // SEPT14 // septin 14 // 7p11...   \n","4      NM_001005277 // OR4F16 // olfactory receptor, ...   \n","...                                                  ...   \n","33292                                                ---   \n","33293                                                ---   \n","33294                                                ---   \n","33295                                                ---   \n","33296                                                ---   \n","\n","                                         mrna_assignment          category  \n","0                                                    ---               ---  \n","1      ENST00000328113 // ENSEMBL // cdna:pseudogene ...              main  \n","2      NM_001005240 // RefSeq // Homo sapiens olfacto...              main  \n","3      ENST00000388975 // ENSEMBL // Septin-14 gene:E...              main  \n","4      NM_001005277 // RefSeq // Homo sapiens olfacto...              main  \n","...                                                  ...               ...  \n","33292  NM_002551.3 // --- // Homo sapiens olfactory r...  flmrna->unmapped  \n","33293  NM_018396.2 // --- // Homo sapiens methyltrans...  flmrna->unmapped  \n","33294  NM_001040145.1 // --- // Homo sapiens chromoso...  flmrna->unmapped  \n","33295                                                ---               ---  \n","33296                                                ---               ---  \n","\n","[33297 rows x 12 columns]"],"text/html":["\n","  <div id=\"df-769a04e9-1445-443c-b20e-0f7ddc0a5588\" class=\"colab-df-container\">\n","    <div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>ID</th>\n","      <th>GB_LIST</th>\n","      <th>SPOT_ID</th>\n","      <th>seqname</th>\n","      <th>RANGE_GB</th>\n","      <th>RANGE_STRAND</th>\n","      <th>RANGE_START</th>\n","      <th>RANGE_STOP</th>\n","      <th>total_probes</th>\n","      <th>gene_assignment</th>\n","      <th>mrna_assignment</th>\n","      <th>category</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>7896736</td>\n","      <td>NaN</td>\n","      <td>chr1:53049-54936</td>\n","      <td>chr1</td>\n","      <td>NC_000001.10</td>\n","      <td>+</td>\n","      <td>53049</td>\n","      <td>54936</td>\n","      <td>7</td>\n","      <td>---</td>\n","      <td>---</td>\n","      <td>---</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>7896738</td>\n","      <td>NaN</td>\n","      <td>chr1:63015-63887</td>\n","      <td>chr1</td>\n","      <td>NC_000001.10</td>\n","      <td>+</td>\n","      <td>63015</td>\n","      <td>63887</td>\n","      <td>31</td>\n","      <td>---</td>\n","      <td>ENST00000328113 // ENSEMBL // cdna:pseudogene ...</td>\n","      <td>main</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>7896740</td>\n","      <td>NM_001005240,NM_001004195,NM_001005484,BC13684...</td>\n","      <td>chr1:69091-70008</td>\n","      <td>chr1</td>\n","      <td>NC_000001.10</td>\n","      <td>+</td>\n","      <td>69091</td>\n","      <td>70008</td>\n","      <td>24</td>\n","      <td>NM_001005240 // OR4F17 // olfactory receptor, ...</td>\n","      <td>NM_001005240 // RefSeq // Homo sapiens olfacto...</td>\n","      <td>main</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>7896742</td>\n","      <td>BC118988,AL137655</td>\n","      <td>chr1:334129-334296</td>\n","      <td>chr1</td>\n","      <td>NC_000001.10</td>\n","      <td>+</td>\n","      <td>334129</td>\n","      <td>334296</td>\n","      <td>6</td>\n","      <td>ENST00000388975 // SEPT14 // septin 14 // 7p11...</td>\n","      <td>ENST00000388975 // ENSEMBL // Septin-14 gene:E...</td>\n","      <td>main</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>7896744</td>\n","      <td>NM_001005277,NM_001005221,NM_001005224,NM_0010...</td>\n","      <td>chr1:367659-368597</td>\n","      <td>chr1</td>\n","      <td>NC_000001.10</td>\n","      <td>+</td>\n","      <td>367659</td>\n","      <td>368597</td>\n","      <td>36</td>\n","      <td>NM_001005277 // OR4F16 // olfactory receptor, ...</td>\n","      <td>NM_001005277 // RefSeq // Homo sapiens olfacto...</td>\n","      <td>main</td>\n","    </tr>\n","    <tr>\n","      <th>...</th>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","    </tr>\n","    <tr>\n","      <th>33292</th>\n","      <td>8180416</td>\n","      <td>NM_002551.3</td>\n","      <td>flmrna-&gt;unmapped</td>\n","      <td>---</td>\n","      <td>NaN</td>\n","      <td>---</td>\n","      <td>0</td>\n","      <td>0</td>\n","      <td>39</td>\n","      <td>---</td>\n","      <td>NM_002551.3 // --- // Homo sapiens olfactory r...</td>\n","      <td>flmrna-&gt;unmapped</td>\n","    </tr>\n","    <tr>\n","      <th>33293</th>\n","      <td>8180417</td>\n","      <td>NM_018396.2</td>\n","      <td>flmrna-&gt;unmapped</td>\n","      <td>---</td>\n","      <td>NaN</td>\n","      <td>---</td>\n","      <td>0</td>\n","      <td>0</td>\n","      <td>33</td>\n","      <td>---</td>\n","      <td>NM_018396.2 // --- // Homo sapiens methyltrans...</td>\n","      <td>flmrna-&gt;unmapped</td>\n","    </tr>\n","    <tr>\n","      <th>33294</th>\n","      <td>8180418</td>\n","      <td>NM_001040145.1</td>\n","      <td>flmrna-&gt;unmapped</td>\n","      <td>---</td>\n","      <td>NaN</td>\n","      <td>---</td>\n","      <td>0</td>\n","      <td>0</td>\n","      <td>6</td>\n","      <td>---</td>\n","      <td>NM_001040145.1 // --- // Homo sapiens chromoso...</td>\n","      <td>flmrna-&gt;unmapped</td>\n","    </tr>\n","    <tr>\n","      <th>33295</th>\n","      <td>7976552</td>\n","      <td>NaN</td>\n","      <td>chr14:96154516-96154596</td>\n","      <td>chr14</td>\n","      <td>NC_000014.8</td>\n","      <td>+</td>\n","      <td>96154516</td>\n","      <td>96154596</td>\n","      <td>4</td>\n","      <td>---</td>\n","      <td>---</td>\n","      <td>---</td>\n","    </tr>\n","    <tr>\n","      <th>33296</th>\n","      <td>8020352</td>\n","      <td>NaN</td>\n","      <td>chr18:14341593-14342513</td>\n","      <td>chr18</td>\n","      <td>NC_000018.9</td>\n","      <td>+</td>\n","      <td>14341593</td>\n","      <td>14342513</td>\n","      <td>9</td>\n","      <td>---</td>\n","      <td>---</td>\n","      <td>---</td>\n","    </tr>\n","  </tbody>\n","</table>\n","<p>33297 rows × 12 columns</p>\n","</div>\n","    <div class=\"colab-df-buttons\">\n","\n","  <div class=\"colab-df-container\">\n","    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-769a04e9-1445-443c-b20e-0f7ddc0a5588')\"\n","            title=\"Convert this dataframe to an interactive table.\"\n","            style=\"display:none;\">\n","\n","  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n","    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n","  </svg>\n","    </button>\n","\n","  <style>\n","    .colab-df-container {\n","      display:flex;\n","      gap: 12px;\n","    }\n","\n","    .colab-df-convert {\n","      background-color: #E8F0FE;\n","      border: none;\n","      border-radius: 50%;\n","      cursor: pointer;\n","      display: none;\n","      fill: #1967D2;\n","      height: 32px;\n","      padding: 0 0 0 0;\n","      width: 32px;\n","    }\n","\n","    .colab-df-convert:hover {\n","      background-color: #E2EBFA;\n","      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n","      fill: #174EA6;\n","    }\n","\n","    .colab-df-buttons div {\n","      margin-bottom: 4px;\n","    }\n","\n","    [theme=dark] .colab-df-convert {\n","      background-color: #3B4455;\n","      fill: #D2E3FC;\n","    }\n","\n","    [theme=dark] .colab-df-convert:hover {\n","      background-color: #434B5C;\n","      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n","      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n","      fill: #FFFFFF;\n","    }\n","  </style>\n","\n","    <script>\n","      const buttonEl =\n","        document.querySelector('#df-769a04e9-1445-443c-b20e-0f7ddc0a5588 button.colab-df-convert');\n","      buttonEl.style.display =\n","        google.colab.kernel.accessAllowed ? 'block' : 'none';\n","\n","      async function convertToInteractive(key) {\n","        const element = document.querySelector('#df-769a04e9-1445-443c-b20e-0f7ddc0a5588');\n","        const dataTable =\n","          await google.colab.kernel.invokeFunction('convertToInteractive',\n","                                                    [key], {});\n","        if (!dataTable) return;\n","\n","        const docLinkHtml = 'Like what you see? Visit the ' +\n","          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n","          + ' to learn more about interactive tables.';\n","        element.innerHTML = '';\n","        dataTable['output_type'] = 'display_data';\n","        await google.colab.output.renderOutput(dataTable, element);\n","        const docLink = document.createElement('div');\n","        docLink.innerHTML = docLinkHtml;\n","        element.appendChild(docLink);\n","      }\n","    </script>\n","  </div>\n","\n","\n","    <div id=\"df-83dc0bc5-459f-4502-9dd6-cb7f6279d9db\">\n","      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-83dc0bc5-459f-4502-9dd6-cb7f6279d9db')\"\n","                title=\"Suggest charts\"\n","                style=\"display:none;\">\n","\n","<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n","     width=\"24px\">\n","    <g>\n","        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n","    </g>\n","</svg>\n","      </button>\n","\n","<style>\n","  .colab-df-quickchart {\n","      --bg-color: #E8F0FE;\n","      --fill-color: #1967D2;\n","      --hover-bg-color: #E2EBFA;\n","      --hover-fill-color: #174EA6;\n","      --disabled-fill-color: #AAA;\n","      --disabled-bg-color: #DDD;\n","  }\n","\n","  [theme=dark] .colab-df-quickchart {\n","      --bg-color: #3B4455;\n","      --fill-color: #D2E3FC;\n","      --hover-bg-color: #434B5C;\n","      --hover-fill-color: #FFFFFF;\n","      --disabled-bg-color: #3B4455;\n","      --disabled-fill-color: #666;\n","  }\n","\n","  .colab-df-quickchart {\n","    background-color: var(--bg-color);\n","    border: none;\n","    border-radius: 50%;\n","    cursor: pointer;\n","    display: none;\n","    fill: var(--fill-color);\n","    height: 32px;\n","    padding: 0;\n","    width: 32px;\n","  }\n","\n","  .colab-df-quickchart:hover {\n","    background-color: var(--hover-bg-color);\n","    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n","    fill: var(--button-hover-fill-color);\n","  }\n","\n","  .colab-df-quickchart-complete:disabled,\n","  .colab-df-quickchart-complete:disabled:hover {\n","    background-color: var(--disabled-bg-color);\n","    fill: var(--disabled-fill-color);\n","    box-shadow: none;\n","  }\n","\n","  .colab-df-spinner {\n","    border: 2px solid var(--fill-color);\n","    border-color: transparent;\n","    border-bottom-color: var(--fill-color);\n","    animation:\n","      spin 1s steps(1) infinite;\n","  }\n","\n","  @keyframes spin {\n","    0% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","      border-left-color: var(--fill-color);\n","    }\n","    20% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    30% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","      border-right-color: var(--fill-color);\n","    }\n","    40% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    60% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","    }\n","    80% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-bottom-color: var(--fill-color);\n","    }\n","    90% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","    }\n","  }\n","</style>\n","\n","      <script>\n","        async function quickchart(key) {\n","          const quickchartButtonEl =\n","            document.querySelector('#' + key + ' button');\n","          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n","          quickchartButtonEl.classList.add('colab-df-spinner');\n","          try {\n","            const charts = await google.colab.kernel.invokeFunction(\n","                'suggestCharts', [key], {});\n","          } catch (error) {\n","            console.error('Error during call to suggestCharts:', error);\n","          }\n","          quickchartButtonEl.classList.remove('colab-df-spinner');\n","          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n","        }\n","        (() => {\n","          let quickchartButtonEl =\n","            document.querySelector('#df-83dc0bc5-459f-4502-9dd6-cb7f6279d9db button');\n","          quickchartButtonEl.style.display =\n","            google.colab.kernel.accessAllowed ? 'block' : 'none';\n","        })();\n","      </script>\n","    </div>\n","\n","  <div id=\"id_e2f3e249-554d-4b79-ac8b-53c85bd9e249\">\n","    <style>\n","      .colab-df-generate {\n","        background-color: #E8F0FE;\n","        border: none;\n","        border-radius: 50%;\n","        cursor: pointer;\n","        display: none;\n","        fill: #1967D2;\n","        height: 32px;\n","        padding: 0 0 0 0;\n","        width: 32px;\n","      }\n","\n","      .colab-df-generate:hover {\n","        background-color: #E2EBFA;\n","        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n","        fill: #174EA6;\n","      }\n","\n","      [theme=dark] .colab-df-generate {\n","        background-color: #3B4455;\n","        fill: #D2E3FC;\n","      }\n","\n","      [theme=dark] .colab-df-generate:hover {\n","        background-color: #434B5C;\n","        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n","        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n","        fill: #FFFFFF;\n","      }\n","    </style>\n","    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('GPL11532_32230_df')\"\n","            title=\"Generate code using this dataframe.\"\n","            style=\"display:none;\">\n","\n","  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n","       width=\"24px\">\n","    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n","  </svg>\n","    </button>\n","    <script>\n","      (() => {\n","      const buttonEl =\n","        document.querySelector('#id_e2f3e249-554d-4b79-ac8b-53c85bd9e249 button.colab-df-generate');\n","      buttonEl.style.display =\n","        google.colab.kernel.accessAllowed ? 'block' : 'none';\n","\n","      buttonEl.onclick = () => {\n","        google.colab.notebook.generateWithVariable('GPL11532_32230_df');\n","      }\n","      })();\n","    </script>\n","  </div>\n","\n","    </div>\n","  </div>\n"],"application/vnd.google.colaboratory.intrinsic+json":{"type":"dataframe","variable_name":"GPL11532_32230_df","summary":"{\n  \"name\": \"GPL11532_32230_df\",\n  \"rows\": 33297,\n  \"fields\": [\n    {\n      \"column\": \"ID\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 91844,\n        \"min\": 7892501,\n        \"max\": 8180418,\n        \"num_unique_values\": 33297,\n        \"samples\": [\n          8027285,\n          8156873,\n          7894170\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"GB_LIST\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 21919,\n        \"samples\": [\n          \"NM_007174,AY681966\",\n          \"NM_012418,NM_001077182,AF030165\",\n          \"AB075866,BC035704,XM_001723010,XM_003119731,XM_003120950\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"SPOT_ID\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 28694,\n        \"samples\": [\n          \"chr11:64658827-64658911\",\n          \"chr17:17921334-17942480\",\n          \"chr19:52534632-52552074\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"seqname\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 35,\n        \"samples\": [\n          \"chr5\",\n          \"chr18\",\n          \"chr4_gl000193_random\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"RANGE_GB\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 25,\n        \"samples\": [\n          \"NC_000017.10\",\n          \"NC_000004.11\",\n          \"NC_000001.10\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"RANGE_STRAND\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 3,\n        \"samples\": [\n          \"+\",\n          \"-\",\n          \"---\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"RANGE_START\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 57905488,\n        \"min\": 0,\n        \"max\": 249230899,\n        \"num_unique_values\": 28646,\n        \"samples\": [\n          76751934,\n          20492606,\n          68702960\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"RANGE_STOP\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 57914563,\n        \"min\": 0,\n        \"max\": 249231268,\n        \"num_unique_values\": 28613,\n        \"samples\": [\n          38269042,\n          33647651,\n          12035595\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"total_probes\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 22,\n        \"min\": 1,\n        \"max\": 1189,\n        \"num_unique_values\": 178,\n        \"samples\": [\n          41,\n          34,\n          2\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"gene_assignment\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 21479,\n        \"samples\": [\n          \"NM_004048 // B2M // beta-2-microglobulin // 15q21-q22.2 // 567 /// ENST00000349264 // B2M // beta-2-microglobulin // 15q21-q22.2 // 567 /// BC032589 // B2M // beta-2-microglobulin // 15q21-q22.2 // 567 /// ENST00000396754 // B2M // beta-2-microglobulin // 15q21-q22.2 // 567\",\n          \"NM_004767 // GPR37L1 // G protein-coupled receptor 37 like 1 // 1q32.1 // 9283 /// ENST00000367282 // GPR37L1 // G protein-coupled receptor 37 like 1 // 1q32.1 // 9283 /// BC050334 // GPR37L1 // G protein-coupled receptor 37 like 1 // 1q32.1 // 9283\",\n          \"NM_014014 // SNRNP200 // small nuclear ribonucleoprotein 200kDa (U5) // 2q11.2 // 23020 /// ENST00000323853 // SNRNP200 // small nuclear ribonucleoprotein 200kDa (U5) // 2q11.2 // 23020 /// AY572488 // SNRNP200 // small nuclear ribonucleoprotein 200kDa (U5) // 2q11.2 // 23020 /// ENST00000349783 // SNRNP200 // small nuclear ribonucleoprotein 200kDa (U5) // 2q11.2 // 23020 /// ENST00000429650 // SNRNP200 // small nuclear ribonucleoprotein 200kDa (U5) // 2q11.2 // 23020\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"mrna_assignment\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 28370,\n        \"samples\": [\n          \"ENST00000458753 // ENSEMBL // ncrna:snoRNA chromosome:GRCh37:5:118404992:118405095:-1 gene:ENSG00000239011 // chr5 // 100 // 100 // 25 // 25 // 0\",\n          \"NM_012367 // RefSeq // Homo sapiens olfactory receptor, family 2, subfamily B, member 6 (OR2B6), mRNA. // chr6 // 100 // 100 // 21 // 21 // 0 /// ENST00000244623 // ENSEMBL // Olfactory receptor 2B6 gene:ENSG00000124657 // chr6 // 100 // 100 // 21 // 21 // 0 /// BC110457 // GenBank // Homo sapiens olfactory receptor, family 2, subfamily B, member 6, mRNA (cDNA clone MGC:120190 IMAGE:40022996), complete cds. // chr6 // 100 // 100 // 21 // 21 // 0\",\n          \"NR_029618 // RefSeq // Homo sapiens microRNA 199a-2 (MIR199A2), microRNA. // chr1 // 100 // 100 // 25 // 25 // 0 /// hsa-mir-199a-2 // miRBase Micro RNA Database // MI0000281 Homo sapiens miR-199a-2 stem-loop // chr1 // 100 // 100 // 25 // 25 // 0\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"category\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 7,\n        \"samples\": [\n          \"---\",\n          \"main\",\n          \"normgene->intron\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"}},"metadata":{},"execution_count":16}]},{"cell_type":"markdown","source":["| 컬럼명                  | 의미 설명                                                               |\n","| -------------------- | ------------------------------------------------------------------- |\n","| **ID**               | probe ID (예: `7896738`, `7896817`) → `series matrix`의 `ID_REF`와 매칭됨 |\n","| **GB\\_LIST**         | GenBank accession number (예: `NM_001005240`, `BC118988` 등)          |\n","| **SPOT\\_ID**         | 위치 정보 (예: `chr1:901877-910488`)                                     |\n","| **seqname**          | 해당 유전자가 위치한 염색체명 (예: `chr1`, `chr14`)                               |\n","| **RANGE\\_GB**        | GenBank 염색체 위치 정보 (예: `NC_000001.10`)                               |\n","| **RANGE\\_STRAND**    | DNA 가닥 방향 (`+` 또는 `-`)                                              |\n","| **RANGE\\_START**     | 유전자 시작 위치 (예: `53049`)                                              |\n","| **RANGE\\_STOP**      | 유전자 종료 위치 (예: `54936`)                                              |\n","| **total\\_probes**    | 해당 probe에 할당된 총 탐침 수                                                |\n","| **gene\\_assignment** | 유전자 주석 정보. `//` 사이에 유전자 이름이 들어 있으며, 복수 가능                           |\n","| **mrna\\_assignment** | mRNA 정보 (`ENSEMBL`, `RefSeq`, `GenBank` 등)                          |\n","| **category**         | 주석 타입 (예: `main`, `flmrna->unmapped` 등)                             |\n"],"metadata":{"id":"Wvc6aLMrABUV"}},{"cell_type":"code","source":["import re\n","\n","# 예시 DataFrame: GPL11532_32230_df\n","# gene_assignment에서 유전자 이름 추출\n","def extract_gene_symbol(text):\n","    if pd.isna(text):\n","        return None\n","    match = re.search(r'//\\s*(.*?)\\s*//', text)\n","    return match.group(1) if match else None\n","\n","# 새 열 추가\n","platform_df= GPL11532_32230_df.copy()\n","platform_df['gene_symbol'] = GPL11532_32230_df['gene_assignment'].apply(extract_gene_symbol)"],"metadata":{"id":"acNW7u7XCI3P","executionInfo":{"status":"ok","timestamp":1750032137922,"user_tz":-540,"elapsed":56,"user":{"displayName":"권혜빈","userId":"01947391368488800076"}}},"execution_count":17,"outputs":[]},{"cell_type":"code","source":["platform_df"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":825},"id":"uIZwTzohCR1V","executionInfo":{"status":"ok","timestamp":1750032143063,"user_tz":-540,"elapsed":556,"user":{"displayName":"권혜빈","userId":"01947391368488800076"}},"outputId":"272a3abd-6eb2-45df-f92f-549abaf15469"},"execution_count":18,"outputs":[{"output_type":"execute_result","data":{"text/plain":["            ID                                            GB_LIST  \\\n","0      7896736                                                NaN   \n","1      7896738                                                NaN   \n","2      7896740  NM_001005240,NM_001004195,NM_001005484,BC13684...   \n","3      7896742                                  BC118988,AL137655   \n","4      7896744  NM_001005277,NM_001005221,NM_001005224,NM_0010...   \n","...        ...                                                ...   \n","33292  8180416                                        NM_002551.3   \n","33293  8180417                                        NM_018396.2   \n","33294  8180418                                     NM_001040145.1   \n","33295  7976552                                                NaN   \n","33296  8020352                                                NaN   \n","\n","                       SPOT_ID seqname      RANGE_GB RANGE_STRAND  \\\n","0             chr1:53049-54936    chr1  NC_000001.10            +   \n","1             chr1:63015-63887    chr1  NC_000001.10            +   \n","2             chr1:69091-70008    chr1  NC_000001.10            +   \n","3           chr1:334129-334296    chr1  NC_000001.10            +   \n","4           chr1:367659-368597    chr1  NC_000001.10            +   \n","...                        ...     ...           ...          ...   \n","33292         flmrna->unmapped     ---           NaN          ---   \n","33293         flmrna->unmapped     ---           NaN          ---   \n","33294         flmrna->unmapped     ---           NaN          ---   \n","33295  chr14:96154516-96154596   chr14   NC_000014.8            +   \n","33296  chr18:14341593-14342513   chr18   NC_000018.9            +   \n","\n","       RANGE_START  RANGE_STOP  total_probes  \\\n","0            53049       54936             7   \n","1            63015       63887            31   \n","2            69091       70008            24   \n","3           334129      334296             6   \n","4           367659      368597            36   \n","...            ...         ...           ...   \n","33292            0           0            39   \n","33293            0           0            33   \n","33294            0           0             6   \n","33295     96154516    96154596             4   \n","33296     14341593    14342513             9   \n","\n","                                         gene_assignment  \\\n","0                                                    ---   \n","1                                                    ---   \n","2      NM_001005240 // OR4F17 // olfactory receptor, ...   \n","3      ENST00000388975 // SEPT14 // septin 14 // 7p11...   \n","4      NM_001005277 // OR4F16 // olfactory receptor, ...   \n","...                                                  ...   \n","33292                                                ---   \n","33293                                                ---   \n","33294                                                ---   \n","33295                                                ---   \n","33296                                                ---   \n","\n","                                         mrna_assignment          category  \\\n","0                                                    ---               ---   \n","1      ENST00000328113 // ENSEMBL // cdna:pseudogene ...              main   \n","2      NM_001005240 // RefSeq // Homo sapiens olfacto...              main   \n","3      ENST00000388975 // ENSEMBL // Septin-14 gene:E...              main   \n","4      NM_001005277 // RefSeq // Homo sapiens olfacto...              main   \n","...                                                  ...               ...   \n","33292  NM_002551.3 // --- // Homo sapiens olfactory r...  flmrna->unmapped   \n","33293  NM_018396.2 // --- // Homo sapiens methyltrans...  flmrna->unmapped   \n","33294  NM_001040145.1 // --- // Homo sapiens chromoso...  flmrna->unmapped   \n","33295                                                ---               ---   \n","33296                                                ---               ---   \n","\n","      gene_symbol  \n","0            None  \n","1            None  \n","2          OR4F17  \n","3          SEPT14  \n","4          OR4F16  \n","...           ...  \n","33292        None  \n","33293        None  \n","33294        None  \n","33295        None  \n","33296        None  \n","\n","[33297 rows x 13 columns]"],"text/html":["\n","  <div id=\"df-8574784d-af4a-4e41-a454-643fdadf8c12\" class=\"colab-df-container\">\n","    <div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>ID</th>\n","      <th>GB_LIST</th>\n","      <th>SPOT_ID</th>\n","      <th>seqname</th>\n","      <th>RANGE_GB</th>\n","      <th>RANGE_STRAND</th>\n","      <th>RANGE_START</th>\n","      <th>RANGE_STOP</th>\n","      <th>total_probes</th>\n","      <th>gene_assignment</th>\n","      <th>mrna_assignment</th>\n","      <th>category</th>\n","      <th>gene_symbol</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>7896736</td>\n","      <td>NaN</td>\n","      <td>chr1:53049-54936</td>\n","      <td>chr1</td>\n","      <td>NC_000001.10</td>\n","      <td>+</td>\n","      <td>53049</td>\n","      <td>54936</td>\n","      <td>7</td>\n","      <td>---</td>\n","      <td>---</td>\n","      <td>---</td>\n","      <td>None</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>7896738</td>\n","      <td>NaN</td>\n","      <td>chr1:63015-63887</td>\n","      <td>chr1</td>\n","      <td>NC_000001.10</td>\n","      <td>+</td>\n","      <td>63015</td>\n","      <td>63887</td>\n","      <td>31</td>\n","      <td>---</td>\n","      <td>ENST00000328113 // ENSEMBL // cdna:pseudogene ...</td>\n","      <td>main</td>\n","      <td>None</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>7896740</td>\n","      <td>NM_001005240,NM_001004195,NM_001005484,BC13684...</td>\n","      <td>chr1:69091-70008</td>\n","      <td>chr1</td>\n","      <td>NC_000001.10</td>\n","      <td>+</td>\n","      <td>69091</td>\n","      <td>70008</td>\n","      <td>24</td>\n","      <td>NM_001005240 // OR4F17 // olfactory receptor, ...</td>\n","      <td>NM_001005240 // RefSeq // Homo sapiens olfacto...</td>\n","      <td>main</td>\n","      <td>OR4F17</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>7896742</td>\n","      <td>BC118988,AL137655</td>\n","      <td>chr1:334129-334296</td>\n","      <td>chr1</td>\n","      <td>NC_000001.10</td>\n","      <td>+</td>\n","      <td>334129</td>\n","      <td>334296</td>\n","      <td>6</td>\n","      <td>ENST00000388975 // SEPT14 // septin 14 // 7p11...</td>\n","      <td>ENST00000388975 // ENSEMBL // Septin-14 gene:E...</td>\n","      <td>main</td>\n","      <td>SEPT14</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>7896744</td>\n","      <td>NM_001005277,NM_001005221,NM_001005224,NM_0010...</td>\n","      <td>chr1:367659-368597</td>\n","      <td>chr1</td>\n","      <td>NC_000001.10</td>\n","      <td>+</td>\n","      <td>367659</td>\n","      <td>368597</td>\n","      <td>36</td>\n","      <td>NM_001005277 // OR4F16 // olfactory receptor, ...</td>\n","      <td>NM_001005277 // RefSeq // Homo sapiens olfacto...</td>\n","      <td>main</td>\n","      <td>OR4F16</td>\n","    </tr>\n","    <tr>\n","      <th>...</th>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","    </tr>\n","    <tr>\n","      <th>33292</th>\n","      <td>8180416</td>\n","      <td>NM_002551.3</td>\n","      <td>flmrna-&gt;unmapped</td>\n","      <td>---</td>\n","      <td>NaN</td>\n","      <td>---</td>\n","      <td>0</td>\n","      <td>0</td>\n","      <td>39</td>\n","      <td>---</td>\n","      <td>NM_002551.3 // --- // Homo sapiens olfactory r...</td>\n","      <td>flmrna-&gt;unmapped</td>\n","      <td>None</td>\n","    </tr>\n","    <tr>\n","      <th>33293</th>\n","      <td>8180417</td>\n","      <td>NM_018396.2</td>\n","      <td>flmrna-&gt;unmapped</td>\n","      <td>---</td>\n","      <td>NaN</td>\n","      <td>---</td>\n","      <td>0</td>\n","      <td>0</td>\n","      <td>33</td>\n","      <td>---</td>\n","      <td>NM_018396.2 // --- // Homo sapiens methyltrans...</td>\n","      <td>flmrna-&gt;unmapped</td>\n","      <td>None</td>\n","    </tr>\n","    <tr>\n","      <th>33294</th>\n","      <td>8180418</td>\n","      <td>NM_001040145.1</td>\n","      <td>flmrna-&gt;unmapped</td>\n","      <td>---</td>\n","      <td>NaN</td>\n","      <td>---</td>\n","      <td>0</td>\n","      <td>0</td>\n","      <td>6</td>\n","      <td>---</td>\n","      <td>NM_001040145.1 // --- // Homo sapiens chromoso...</td>\n","      <td>flmrna-&gt;unmapped</td>\n","      <td>None</td>\n","    </tr>\n","    <tr>\n","      <th>33295</th>\n","      <td>7976552</td>\n","      <td>NaN</td>\n","      <td>chr14:96154516-96154596</td>\n","      <td>chr14</td>\n","      <td>NC_000014.8</td>\n","      <td>+</td>\n","      <td>96154516</td>\n","      <td>96154596</td>\n","      <td>4</td>\n","      <td>---</td>\n","      <td>---</td>\n","      <td>---</td>\n","      <td>None</td>\n","    </tr>\n","    <tr>\n","      <th>33296</th>\n","      <td>8020352</td>\n","      <td>NaN</td>\n","      <td>chr18:14341593-14342513</td>\n","      <td>chr18</td>\n","      <td>NC_000018.9</td>\n","      <td>+</td>\n","      <td>14341593</td>\n","      <td>14342513</td>\n","      <td>9</td>\n","      <td>---</td>\n","      <td>---</td>\n","      <td>---</td>\n","      <td>None</td>\n","    </tr>\n","  </tbody>\n","</table>\n","<p>33297 rows × 13 columns</p>\n","</div>\n","    <div class=\"colab-df-buttons\">\n","\n","  <div class=\"colab-df-container\">\n","    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-8574784d-af4a-4e41-a454-643fdadf8c12')\"\n","            title=\"Convert this dataframe to an interactive table.\"\n","            style=\"display:none;\">\n","\n","  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n","    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n","  </svg>\n","    </button>\n","\n","  <style>\n","    .colab-df-container {\n","      display:flex;\n","      gap: 12px;\n","    }\n","\n","    .colab-df-convert {\n","      background-color: #E8F0FE;\n","      border: none;\n","      border-radius: 50%;\n","      cursor: pointer;\n","      display: none;\n","      fill: #1967D2;\n","      height: 32px;\n","      padding: 0 0 0 0;\n","      width: 32px;\n","    }\n","\n","    .colab-df-convert:hover {\n","      background-color: #E2EBFA;\n","      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n","      fill: #174EA6;\n","    }\n","\n","    .colab-df-buttons div {\n","      margin-bottom: 4px;\n","    }\n","\n","    [theme=dark] .colab-df-convert {\n","      background-color: #3B4455;\n","      fill: #D2E3FC;\n","    }\n","\n","    [theme=dark] .colab-df-convert:hover {\n","      background-color: #434B5C;\n","      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n","      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n","      fill: #FFFFFF;\n","    }\n","  </style>\n","\n","    <script>\n","      const buttonEl =\n","        document.querySelector('#df-8574784d-af4a-4e41-a454-643fdadf8c12 button.colab-df-convert');\n","      buttonEl.style.display =\n","        google.colab.kernel.accessAllowed ? 'block' : 'none';\n","\n","      async function convertToInteractive(key) {\n","        const element = document.querySelector('#df-8574784d-af4a-4e41-a454-643fdadf8c12');\n","        const dataTable =\n","          await google.colab.kernel.invokeFunction('convertToInteractive',\n","                                                    [key], {});\n","        if (!dataTable) return;\n","\n","        const docLinkHtml = 'Like what you see? Visit the ' +\n","          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n","          + ' to learn more about interactive tables.';\n","        element.innerHTML = '';\n","        dataTable['output_type'] = 'display_data';\n","        await google.colab.output.renderOutput(dataTable, element);\n","        const docLink = document.createElement('div');\n","        docLink.innerHTML = docLinkHtml;\n","        element.appendChild(docLink);\n","      }\n","    </script>\n","  </div>\n","\n","\n","    <div id=\"df-b971bef6-9aff-42b3-a8ce-b9b791f04c3a\">\n","      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-b971bef6-9aff-42b3-a8ce-b9b791f04c3a')\"\n","                title=\"Suggest charts\"\n","                style=\"display:none;\">\n","\n","<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n","     width=\"24px\">\n","    <g>\n","        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n","    </g>\n","</svg>\n","      </button>\n","\n","<style>\n","  .colab-df-quickchart {\n","      --bg-color: #E8F0FE;\n","      --fill-color: #1967D2;\n","      --hover-bg-color: #E2EBFA;\n","      --hover-fill-color: #174EA6;\n","      --disabled-fill-color: #AAA;\n","      --disabled-bg-color: #DDD;\n","  }\n","\n","  [theme=dark] .colab-df-quickchart {\n","      --bg-color: #3B4455;\n","      --fill-color: #D2E3FC;\n","      --hover-bg-color: #434B5C;\n","      --hover-fill-color: #FFFFFF;\n","      --disabled-bg-color: #3B4455;\n","      --disabled-fill-color: #666;\n","  }\n","\n","  .colab-df-quickchart {\n","    background-color: var(--bg-color);\n","    border: none;\n","    border-radius: 50%;\n","    cursor: pointer;\n","    display: none;\n","    fill: var(--fill-color);\n","    height: 32px;\n","    padding: 0;\n","    width: 32px;\n","  }\n","\n","  .colab-df-quickchart:hover {\n","    background-color: var(--hover-bg-color);\n","    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n","    fill: var(--button-hover-fill-color);\n","  }\n","\n","  .colab-df-quickchart-complete:disabled,\n","  .colab-df-quickchart-complete:disabled:hover {\n","    background-color: var(--disabled-bg-color);\n","    fill: var(--disabled-fill-color);\n","    box-shadow: none;\n","  }\n","\n","  .colab-df-spinner {\n","    border: 2px solid var(--fill-color);\n","    border-color: transparent;\n","    border-bottom-color: var(--fill-color);\n","    animation:\n","      spin 1s steps(1) infinite;\n","  }\n","\n","  @keyframes spin {\n","    0% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","      border-left-color: var(--fill-color);\n","    }\n","    20% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    30% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","      border-right-color: var(--fill-color);\n","    }\n","    40% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    60% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","    }\n","    80% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-bottom-color: var(--fill-color);\n","    }\n","    90% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","    }\n","  }\n","</style>\n","\n","      <script>\n","        async function quickchart(key) {\n","          const quickchartButtonEl =\n","            document.querySelector('#' + key + ' button');\n","          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n","          quickchartButtonEl.classList.add('colab-df-spinner');\n","          try {\n","            const charts = await google.colab.kernel.invokeFunction(\n","                'suggestCharts', [key], {});\n","          } catch (error) {\n","            console.error('Error during call to suggestCharts:', error);\n","          }\n","          quickchartButtonEl.classList.remove('colab-df-spinner');\n","          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n","        }\n","        (() => {\n","          let quickchartButtonEl =\n","            document.querySelector('#df-b971bef6-9aff-42b3-a8ce-b9b791f04c3a button');\n","          quickchartButtonEl.style.display =\n","            google.colab.kernel.accessAllowed ? 'block' : 'none';\n","        })();\n","      </script>\n","    </div>\n","\n","  <div id=\"id_8ed5420b-8324-4d21-ae43-a65ffee5a432\">\n","    <style>\n","      .colab-df-generate {\n","        background-color: #E8F0FE;\n","        border: none;\n","        border-radius: 50%;\n","        cursor: pointer;\n","        display: none;\n","        fill: #1967D2;\n","        height: 32px;\n","        padding: 0 0 0 0;\n","        width: 32px;\n","      }\n","\n","      .colab-df-generate:hover {\n","        background-color: #E2EBFA;\n","        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n","        fill: #174EA6;\n","      }\n","\n","      [theme=dark] .colab-df-generate {\n","        background-color: #3B4455;\n","        fill: #D2E3FC;\n","      }\n","\n","      [theme=dark] .colab-df-generate:hover {\n","        background-color: #434B5C;\n","        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n","        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n","        fill: #FFFFFF;\n","      }\n","    </style>\n","    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('platform_df')\"\n","            title=\"Generate code using this dataframe.\"\n","            style=\"display:none;\">\n","\n","  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n","       width=\"24px\">\n","    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n","  </svg>\n","    </button>\n","    <script>\n","      (() => {\n","      const buttonEl =\n","        document.querySelector('#id_8ed5420b-8324-4d21-ae43-a65ffee5a432 button.colab-df-generate');\n","      buttonEl.style.display =\n","        google.colab.kernel.accessAllowed ? 'block' : 'none';\n","\n","      buttonEl.onclick = () => {\n","        google.colab.notebook.generateWithVariable('platform_df');\n","      }\n","      })();\n","    </script>\n","  </div>\n","\n","    </div>\n","  </div>\n"],"application/vnd.google.colaboratory.intrinsic+json":{"type":"dataframe","variable_name":"platform_df","summary":"{\n  \"name\": \"platform_df\",\n  \"rows\": 33297,\n  \"fields\": [\n    {\n      \"column\": \"ID\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 91844,\n        \"min\": 7892501,\n        \"max\": 8180418,\n        \"num_unique_values\": 33297,\n        \"samples\": [\n          8027285,\n          8156873,\n          7894170\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"GB_LIST\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 21919,\n        \"samples\": [\n          \"NM_007174,AY681966\",\n          \"NM_012418,NM_001077182,AF030165\",\n          \"AB075866,BC035704,XM_001723010,XM_003119731,XM_003120950\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"SPOT_ID\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 28694,\n        \"samples\": [\n          \"chr11:64658827-64658911\",\n          \"chr17:17921334-17942480\",\n          \"chr19:52534632-52552074\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"seqname\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 35,\n        \"samples\": [\n          \"chr5\",\n          \"chr18\",\n          \"chr4_gl000193_random\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"RANGE_GB\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 25,\n        \"samples\": [\n          \"NC_000017.10\",\n          \"NC_000004.11\",\n          \"NC_000001.10\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"RANGE_STRAND\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 3,\n        \"samples\": [\n          \"+\",\n          \"-\",\n          \"---\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"RANGE_START\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 57905488,\n        \"min\": 0,\n        \"max\": 249230899,\n        \"num_unique_values\": 28646,\n        \"samples\": [\n          76751934,\n          20492606,\n          68702960\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"RANGE_STOP\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 57914563,\n        \"min\": 0,\n        \"max\": 249231268,\n        \"num_unique_values\": 28613,\n        \"samples\": [\n          38269042,\n          33647651,\n          12035595\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"total_probes\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 22,\n        \"min\": 1,\n        \"max\": 1189,\n        \"num_unique_values\": 178,\n        \"samples\": [\n          41,\n          34,\n          2\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"gene_assignment\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 21479,\n        \"samples\": [\n          \"NM_004048 // B2M // beta-2-microglobulin // 15q21-q22.2 // 567 /// ENST00000349264 // B2M // beta-2-microglobulin // 15q21-q22.2 // 567 /// BC032589 // B2M // beta-2-microglobulin // 15q21-q22.2 // 567 /// ENST00000396754 // B2M // beta-2-microglobulin // 15q21-q22.2 // 567\",\n          \"NM_004767 // GPR37L1 // G protein-coupled receptor 37 like 1 // 1q32.1 // 9283 /// ENST00000367282 // GPR37L1 // G protein-coupled receptor 37 like 1 // 1q32.1 // 9283 /// BC050334 // GPR37L1 // G protein-coupled receptor 37 like 1 // 1q32.1 // 9283\",\n          \"NM_014014 // SNRNP200 // small nuclear ribonucleoprotein 200kDa (U5) // 2q11.2 // 23020 /// ENST00000323853 // SNRNP200 // small nuclear ribonucleoprotein 200kDa (U5) // 2q11.2 // 23020 /// AY572488 // SNRNP200 // small nuclear ribonucleoprotein 200kDa (U5) // 2q11.2 // 23020 /// ENST00000349783 // SNRNP200 // small nuclear ribonucleoprotein 200kDa (U5) // 2q11.2 // 23020 /// ENST00000429650 // SNRNP200 // small nuclear ribonucleoprotein 200kDa (U5) // 2q11.2 // 23020\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"mrna_assignment\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 28370,\n        \"samples\": [\n          \"ENST00000458753 // ENSEMBL // ncrna:snoRNA chromosome:GRCh37:5:118404992:118405095:-1 gene:ENSG00000239011 // chr5 // 100 // 100 // 25 // 25 // 0\",\n          \"NM_012367 // RefSeq // Homo sapiens olfactory receptor, family 2, subfamily B, member 6 (OR2B6), mRNA. // chr6 // 100 // 100 // 21 // 21 // 0 /// ENST00000244623 // ENSEMBL // Olfactory receptor 2B6 gene:ENSG00000124657 // chr6 // 100 // 100 // 21 // 21 // 0 /// BC110457 // GenBank // Homo sapiens olfactory receptor, family 2, subfamily B, member 6, mRNA (cDNA clone MGC:120190 IMAGE:40022996), complete cds. // chr6 // 100 // 100 // 21 // 21 // 0\",\n          \"NR_029618 // RefSeq // Homo sapiens microRNA 199a-2 (MIR199A2), microRNA. // chr1 // 100 // 100 // 25 // 25 // 0 /// hsa-mir-199a-2 // miRBase Micro RNA Database // MI0000281 Homo sapiens miR-199a-2 stem-loop // chr1 // 100 // 100 // 25 // 25 // 0\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"category\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 7,\n        \"samples\": [\n          \"---\",\n          \"main\",\n          \"normgene->intron\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"gene_symbol\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 20254,\n        \"samples\": [\n          \"SLC29A1\",\n          \"ACSM3\",\n          \"SNORD45A\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"}},"metadata":{},"execution_count":18}]},{"cell_type":"code","source":["platform_df.to_csv(\"/content/drive/MyDrive/BIO_MEDICAL_PROJECT/obesity_gene_analysis/data/platform_df.csv\")"],"metadata":{"id":"4mDkYuA0HqoN","executionInfo":{"status":"ok","timestamp":1750033453979,"user_tz":-540,"elapsed":2037,"user":{"displayName":"권혜빈","userId":"01947391368488800076"}}},"execution_count":22,"outputs":[]},{"cell_type":"markdown","source":["# Metadata_df 데이터프레임"],"metadata":{"id":"rqHzp8juDlg4"}},{"cell_type":"code","source":["!pip install GEOparse"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"c02XNFYj98Bz","executionInfo":{"status":"ok","timestamp":1750030910070,"user_tz":-540,"elapsed":9927,"user":{"displayName":"권혜빈","userId":"01947391368488800076"}},"outputId":"40020097-ab11-4930-e5d5-38239de2782d"},"execution_count":12,"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting GEOparse\n","  Downloading GEOparse-2.0.4-py3-none-any.whl.metadata (6.5 kB)\n","Requirement already satisfied: numpy>=1.7 in /usr/local/lib/python3.11/dist-packages (from GEOparse) (2.0.2)\n","Requirement already satisfied: pandas>=0.17 in /usr/local/lib/python3.11/dist-packages (from GEOparse) (2.2.2)\n","Requirement already satisfied: requests>=2.21.0 in /usr/local/lib/python3.11/dist-packages (from GEOparse) (2.32.3)\n","Requirement already satisfied: tqdm>=4.31.1 in /usr/local/lib/python3.11/dist-packages (from GEOparse) (4.67.1)\n","Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.11/dist-packages (from pandas>=0.17->GEOparse) (2.9.0.post0)\n","Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.11/dist-packages (from pandas>=0.17->GEOparse) (2025.2)\n","Requirement already satisfied: tzdata>=2022.7 in /usr/local/lib/python3.11/dist-packages (from pandas>=0.17->GEOparse) (2025.2)\n","Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.11/dist-packages (from requests>=2.21.0->GEOparse) (3.4.2)\n","Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.11/dist-packages (from requests>=2.21.0->GEOparse) (3.10)\n","Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.11/dist-packages (from requests>=2.21.0->GEOparse) (2.4.0)\n","Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.11/dist-packages (from requests>=2.21.0->GEOparse) (2025.4.26)\n","Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.11/dist-packages (from python-dateutil>=2.8.2->pandas>=0.17->GEOparse) (1.17.0)\n","Downloading GEOparse-2.0.4-py3-none-any.whl (29 kB)\n","Installing collected packages: GEOparse\n","Successfully installed GEOparse-2.0.4\n"]}]},{"cell_type":"code","source":["import GEOparse\n","\n","# GEO accession ID가 \"GSE77962\"인 데이터셋을 다운로드 및 파싱하여 gse 객체 생성\n","# 다운로드된 파일은 ./ 디렉토리에 저장됨\n","gse = GEOparse.get_GEO(\"GSE77962\", destdir=\"./\")\n","\n","# 샘플 메타데이터를 저장할 빈 리스트 생성\n","metadata = []\n","\n","# gse.gsms는 각 GSM(샘플)에 대한 정보를 담고 있는 dictionary\n","for gsm_id, gsm in gse.gsms.items():\n","    # 각 샘플의 GSM ID를 포함한 딕셔너리 생성\n","    sample_info = {\"gsm_id\": gsm_id}\n","\n","    # 샘플의 특성 정보(\"characteristics_ch1\")를 하나씩 순회\n","    for line in gsm.metadata.get(\"characteristics_ch1\", []):\n","        # 특성 정보는 보통 \"key: value\" 형태로 되어 있음\n","        if \":\" in line:\n","            # \":\" 기준으로 key와 value 분리 (최초의 ':'만 기준)\n","            key, value = line.split(\":\", 1)\n","\n","            # key 전처리: 공백 -> 언더스코어, 소문자로 통일, 괄호 제거 등\n","            key = key.strip().lower().replace(\" \", \"_\").replace(\"(\", \"\").replace(\")\", \"\")\n","\n","            # value는 앞뒤 공백 제거\n","            value = value.strip()\n","\n","            # 전처리한 key-value 쌍을 sample_info에 추가\n","            sample_info[key] = value\n","\n","    metadata.append(sample_info)\n","\n","metadata_df = pd.DataFrame(metadata)"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"SuDKBzWy-FrY","executionInfo":{"status":"ok","timestamp":1750033693536,"user_tz":-540,"elapsed":111121,"user":{"displayName":"권혜빈","userId":"01947391368488800076"}},"outputId":"9ccb20c1-2c40-4565-e63a-7cd35b686e11"},"execution_count":23,"outputs":[{"output_type":"stream","name":"stderr","text":["16-Jun-2025 00:26:25 DEBUG utils - Directory ./ already exists. Skipping.\n","DEBUG:GEOparse:Directory ./ already exists. Skipping.\n","16-Jun-2025 00:26:25 INFO GEOparse - Downloading ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE77nnn/GSE77962/soft/GSE77962_family.soft.gz to ./GSE77962_family.soft.gz\n","INFO:GEOparse:Downloading ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE77nnn/GSE77962/soft/GSE77962_family.soft.gz to ./GSE77962_family.soft.gz\n","100%|██████████| 47.5M/47.5M [01:32<00:00, 538kB/s] \n","16-Jun-2025 00:28:01 DEBUG downloader - Size validation passed\n","DEBUG:GEOparse:Size validation passed\n","16-Jun-2025 00:28:01 DEBUG downloader - Moving /tmp/tmpgdh_tuxi to /content/GSE77962_family.soft.gz\n","DEBUG:GEOparse:Moving /tmp/tmpgdh_tuxi to /content/GSE77962_family.soft.gz\n","16-Jun-2025 00:28:01 DEBUG downloader - Successfully downloaded ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE77nnn/GSE77962/soft/GSE77962_family.soft.gz\n","DEBUG:GEOparse:Successfully downloaded ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE77nnn/GSE77962/soft/GSE77962_family.soft.gz\n","16-Jun-2025 00:28:01 INFO GEOparse - Parsing ./GSE77962_family.soft.gz: \n","INFO:GEOparse:Parsing ./GSE77962_family.soft.gz: \n","16-Jun-2025 00:28:01 DEBUG GEOparse - DATABASE: GeoMiame\n","DEBUG:GEOparse:DATABASE: GeoMiame\n","16-Jun-2025 00:28:01 DEBUG GEOparse - SERIES: GSE77962\n","DEBUG:GEOparse:SERIES: GSE77962\n","16-Jun-2025 00:28:01 DEBUG GEOparse - PLATFORM: GPL11532\n","DEBUG:GEOparse:PLATFORM: GPL11532\n","16-Jun-2025 00:28:03 DEBUG GEOparse - SAMPLE: GSM2062466\n","DEBUG:GEOparse:SAMPLE: GSM2062466\n","16-Jun-2025 00:28:03 DEBUG GEOparse - SAMPLE: GSM2062467\n","DEBUG:GEOparse:SAMPLE: GSM2062467\n","16-Jun-2025 00:28:03 DEBUG GEOparse - SAMPLE: GSM2062468\n","DEBUG:GEOparse:SAMPLE: GSM2062468\n","16-Jun-2025 00:28:04 DEBUG GEOparse - SAMPLE: GSM2062469\n","DEBUG:GEOparse:SAMPLE: GSM2062469\n","16-Jun-2025 00:28:04 DEBUG GEOparse - SAMPLE: GSM2062470\n","DEBUG:GEOparse:SAMPLE: GSM2062470\n","16-Jun-2025 00:28:04 DEBUG GEOparse - SAMPLE: GSM2062471\n","DEBUG:GEOparse:SAMPLE: GSM2062471\n","16-Jun-2025 00:28:04 DEBUG GEOparse - SAMPLE: GSM2062472\n","DEBUG:GEOparse:SAMPLE: GSM2062472\n","16-Jun-2025 00:28:04 DEBUG GEOparse - SAMPLE: GSM2062473\n","DEBUG:GEOparse:SAMPLE: GSM2062473\n","16-Jun-2025 00:28:04 DEBUG GEOparse - SAMPLE: GSM2062474\n","DEBUG:GEOparse:SAMPLE: GSM2062474\n","16-Jun-2025 00:28:04 DEBUG GEOparse - SAMPLE: GSM2062475\n","DEBUG:GEOparse:SAMPLE: GSM2062475\n","16-Jun-2025 00:28:04 DEBUG GEOparse - SAMPLE: GSM2062476\n","DEBUG:GEOparse:SAMPLE: GSM2062476\n","16-Jun-2025 00:28:04 DEBUG GEOparse - SAMPLE: GSM2062477\n","DEBUG:GEOparse:SAMPLE: GSM2062477\n","16-Jun-2025 00:28:04 DEBUG GEOparse - SAMPLE: GSM2062478\n","DEBUG:GEOparse:SAMPLE: GSM2062478\n","16-Jun-2025 00:28:04 DEBUG GEOparse - SAMPLE: GSM2062479\n","DEBUG:GEOparse:SAMPLE: GSM2062479\n","16-Jun-2025 00:28:04 DEBUG GEOparse - SAMPLE: GSM2062480\n","DEBUG:GEOparse:SAMPLE: GSM2062480\n","16-Jun-2025 00:28:04 DEBUG GEOparse - SAMPLE: GSM2062481\n","DEBUG:GEOparse:SAMPLE: GSM2062481\n","16-Jun-2025 00:28:04 DEBUG GEOparse - SAMPLE: GSM2062482\n","DEBUG:GEOparse:SAMPLE: GSM2062482\n","16-Jun-2025 00:28:04 DEBUG GEOparse - SAMPLE: GSM2062483\n","DEBUG:GEOparse:SAMPLE: GSM2062483\n","16-Jun-2025 00:28:05 DEBUG GEOparse - SAMPLE: GSM2062484\n","DEBUG:GEOparse:SAMPLE: GSM2062484\n","16-Jun-2025 00:28:05 DEBUG GEOparse - SAMPLE: GSM2062485\n","DEBUG:GEOparse:SAMPLE: GSM2062485\n","16-Jun-2025 00:28:05 DEBUG GEOparse - SAMPLE: GSM2062486\n","DEBUG:GEOparse:SAMPLE: GSM2062486\n","16-Jun-2025 00:28:05 DEBUG GEOparse - SAMPLE: GSM2062487\n","DEBUG:GEOparse:SAMPLE: GSM2062487\n","16-Jun-2025 00:28:05 DEBUG GEOparse - SAMPLE: GSM2062488\n","DEBUG:GEOparse:SAMPLE: GSM2062488\n","16-Jun-2025 00:28:05 DEBUG GEOparse - SAMPLE: GSM2062489\n","DEBUG:GEOparse:SAMPLE: GSM2062489\n","16-Jun-2025 00:28:05 DEBUG GEOparse - SAMPLE: GSM2062490\n","DEBUG:GEOparse:SAMPLE: GSM2062490\n","16-Jun-2025 00:28:05 DEBUG GEOparse - SAMPLE: GSM2062491\n","DEBUG:GEOparse:SAMPLE: GSM2062491\n","16-Jun-2025 00:28:05 DEBUG GEOparse - SAMPLE: GSM2062492\n","DEBUG:GEOparse:SAMPLE: GSM2062492\n","16-Jun-2025 00:28:05 DEBUG GEOparse - SAMPLE: GSM2062493\n","DEBUG:GEOparse:SAMPLE: GSM2062493\n","16-Jun-2025 00:28:05 DEBUG GEOparse - SAMPLE: GSM2062494\n","DEBUG:GEOparse:SAMPLE: GSM2062494\n","16-Jun-2025 00:28:05 DEBUG GEOparse - SAMPLE: GSM2062495\n","DEBUG:GEOparse:SAMPLE: GSM2062495\n","16-Jun-2025 00:28:05 DEBUG GEOparse - SAMPLE: GSM2062496\n","DEBUG:GEOparse:SAMPLE: GSM2062496\n","16-Jun-2025 00:28:05 DEBUG GEOparse - SAMPLE: GSM2062497\n","DEBUG:GEOparse:SAMPLE: GSM2062497\n","16-Jun-2025 00:28:06 DEBUG GEOparse - SAMPLE: GSM2062498\n","DEBUG:GEOparse:SAMPLE: GSM2062498\n","16-Jun-2025 00:28:06 DEBUG GEOparse - SAMPLE: GSM2062499\n","DEBUG:GEOparse:SAMPLE: GSM2062499\n","16-Jun-2025 00:28:06 DEBUG GEOparse - SAMPLE: GSM2062500\n","DEBUG:GEOparse:SAMPLE: GSM2062500\n","16-Jun-2025 00:28:06 DEBUG GEOparse - SAMPLE: GSM2062501\n","DEBUG:GEOparse:SAMPLE: GSM2062501\n","16-Jun-2025 00:28:06 DEBUG GEOparse - SAMPLE: GSM2062502\n","DEBUG:GEOparse:SAMPLE: GSM2062502\n","16-Jun-2025 00:28:06 DEBUG GEOparse - SAMPLE: GSM2062503\n","DEBUG:GEOparse:SAMPLE: GSM2062503\n","16-Jun-2025 00:28:06 DEBUG GEOparse - SAMPLE: GSM2062504\n","DEBUG:GEOparse:SAMPLE: GSM2062504\n","16-Jun-2025 00:28:06 DEBUG GEOparse - SAMPLE: GSM2062505\n","DEBUG:GEOparse:SAMPLE: GSM2062505\n","16-Jun-2025 00:28:06 DEBUG GEOparse - SAMPLE: GSM2062506\n","DEBUG:GEOparse:SAMPLE: GSM2062506\n","16-Jun-2025 00:28:06 DEBUG GEOparse - SAMPLE: GSM2062507\n","DEBUG:GEOparse:SAMPLE: GSM2062507\n","16-Jun-2025 00:28:06 DEBUG GEOparse - SAMPLE: GSM2062508\n","DEBUG:GEOparse:SAMPLE: GSM2062508\n","16-Jun-2025 00:28:06 DEBUG GEOparse - SAMPLE: GSM2062509\n","DEBUG:GEOparse:SAMPLE: GSM2062509\n","16-Jun-2025 00:28:06 DEBUG GEOparse - SAMPLE: GSM2062510\n","DEBUG:GEOparse:SAMPLE: GSM2062510\n","16-Jun-2025 00:28:06 DEBUG GEOparse - SAMPLE: GSM2062511\n","DEBUG:GEOparse:SAMPLE: GSM2062511\n","16-Jun-2025 00:28:06 DEBUG GEOparse - SAMPLE: GSM2062512\n","DEBUG:GEOparse:SAMPLE: GSM2062512\n","16-Jun-2025 00:28:06 DEBUG GEOparse - SAMPLE: GSM2062513\n","DEBUG:GEOparse:SAMPLE: GSM2062513\n","16-Jun-2025 00:28:06 DEBUG GEOparse - SAMPLE: GSM2062514\n","DEBUG:GEOparse:SAMPLE: GSM2062514\n","16-Jun-2025 00:28:07 DEBUG GEOparse - SAMPLE: GSM2062515\n","DEBUG:GEOparse:SAMPLE: GSM2062515\n","16-Jun-2025 00:28:07 DEBUG GEOparse - SAMPLE: GSM2062516\n","DEBUG:GEOparse:SAMPLE: GSM2062516\n","16-Jun-2025 00:28:07 DEBUG GEOparse - SAMPLE: GSM2062517\n","DEBUG:GEOparse:SAMPLE: GSM2062517\n","16-Jun-2025 00:28:07 DEBUG GEOparse - SAMPLE: GSM2062518\n","DEBUG:GEOparse:SAMPLE: GSM2062518\n","16-Jun-2025 00:28:07 DEBUG GEOparse - SAMPLE: GSM2062519\n","DEBUG:GEOparse:SAMPLE: GSM2062519\n","16-Jun-2025 00:28:07 DEBUG GEOparse - SAMPLE: GSM2062520\n","DEBUG:GEOparse:SAMPLE: GSM2062520\n","16-Jun-2025 00:28:07 DEBUG GEOparse - SAMPLE: GSM2062521\n","DEBUG:GEOparse:SAMPLE: GSM2062521\n","16-Jun-2025 00:28:07 DEBUG GEOparse - SAMPLE: GSM2062522\n","DEBUG:GEOparse:SAMPLE: GSM2062522\n","16-Jun-2025 00:28:07 DEBUG GEOparse - SAMPLE: GSM2062523\n","DEBUG:GEOparse:SAMPLE: GSM2062523\n","16-Jun-2025 00:28:07 DEBUG GEOparse - SAMPLE: GSM2062524\n","DEBUG:GEOparse:SAMPLE: GSM2062524\n","16-Jun-2025 00:28:07 DEBUG GEOparse - SAMPLE: GSM2062525\n","DEBUG:GEOparse:SAMPLE: GSM2062525\n","16-Jun-2025 00:28:07 DEBUG GEOparse - SAMPLE: GSM2062526\n","DEBUG:GEOparse:SAMPLE: GSM2062526\n","16-Jun-2025 00:28:07 DEBUG GEOparse - SAMPLE: GSM2062527\n","DEBUG:GEOparse:SAMPLE: GSM2062527\n","16-Jun-2025 00:28:08 DEBUG GEOparse - SAMPLE: GSM2062528\n","DEBUG:GEOparse:SAMPLE: GSM2062528\n","16-Jun-2025 00:28:08 DEBUG GEOparse - SAMPLE: GSM2062529\n","DEBUG:GEOparse:SAMPLE: GSM2062529\n","16-Jun-2025 00:28:08 DEBUG GEOparse - SAMPLE: GSM2062530\n","DEBUG:GEOparse:SAMPLE: GSM2062530\n","16-Jun-2025 00:28:08 DEBUG GEOparse - SAMPLE: GSM2062531\n","DEBUG:GEOparse:SAMPLE: GSM2062531\n","16-Jun-2025 00:28:08 DEBUG GEOparse - SAMPLE: GSM2062532\n","DEBUG:GEOparse:SAMPLE: GSM2062532\n","16-Jun-2025 00:28:08 DEBUG GEOparse - SAMPLE: GSM2062533\n","DEBUG:GEOparse:SAMPLE: GSM2062533\n","16-Jun-2025 00:28:08 DEBUG GEOparse - SAMPLE: GSM2062534\n","DEBUG:GEOparse:SAMPLE: GSM2062534\n","16-Jun-2025 00:28:08 DEBUG GEOparse - SAMPLE: GSM2062535\n","DEBUG:GEOparse:SAMPLE: GSM2062535\n","16-Jun-2025 00:28:08 DEBUG GEOparse - SAMPLE: GSM2062536\n","DEBUG:GEOparse:SAMPLE: GSM2062536\n","16-Jun-2025 00:28:08 DEBUG GEOparse - SAMPLE: GSM2062537\n","DEBUG:GEOparse:SAMPLE: GSM2062537\n","16-Jun-2025 00:28:08 DEBUG GEOparse - SAMPLE: GSM2062538\n","DEBUG:GEOparse:SAMPLE: GSM2062538\n","16-Jun-2025 00:28:08 DEBUG GEOparse - SAMPLE: GSM2062539\n","DEBUG:GEOparse:SAMPLE: GSM2062539\n","16-Jun-2025 00:28:08 DEBUG GEOparse - SAMPLE: GSM2062540\n","DEBUG:GEOparse:SAMPLE: GSM2062540\n","16-Jun-2025 00:28:09 DEBUG GEOparse - SAMPLE: GSM2062541\n","DEBUG:GEOparse:SAMPLE: GSM2062541\n","16-Jun-2025 00:28:09 DEBUG GEOparse - SAMPLE: GSM2062542\n","DEBUG:GEOparse:SAMPLE: GSM2062542\n","16-Jun-2025 00:28:09 DEBUG GEOparse - SAMPLE: GSM2062543\n","DEBUG:GEOparse:SAMPLE: GSM2062543\n","16-Jun-2025 00:28:09 DEBUG GEOparse - SAMPLE: GSM2062544\n","DEBUG:GEOparse:SAMPLE: GSM2062544\n","16-Jun-2025 00:28:09 DEBUG GEOparse - SAMPLE: GSM2062545\n","DEBUG:GEOparse:SAMPLE: GSM2062545\n","16-Jun-2025 00:28:09 DEBUG GEOparse - SAMPLE: GSM2062546\n","DEBUG:GEOparse:SAMPLE: GSM2062546\n","16-Jun-2025 00:28:09 DEBUG GEOparse - SAMPLE: GSM2062547\n","DEBUG:GEOparse:SAMPLE: GSM2062547\n","16-Jun-2025 00:28:09 DEBUG GEOparse - SAMPLE: GSM2062548\n","DEBUG:GEOparse:SAMPLE: GSM2062548\n","16-Jun-2025 00:28:09 DEBUG GEOparse - SAMPLE: GSM2062549\n","DEBUG:GEOparse:SAMPLE: GSM2062549\n","16-Jun-2025 00:28:09 DEBUG GEOparse - SAMPLE: GSM2062550\n","DEBUG:GEOparse:SAMPLE: GSM2062550\n","16-Jun-2025 00:28:09 DEBUG GEOparse - SAMPLE: GSM2062551\n","DEBUG:GEOparse:SAMPLE: GSM2062551\n","16-Jun-2025 00:28:09 DEBUG GEOparse - SAMPLE: GSM2062552\n","DEBUG:GEOparse:SAMPLE: GSM2062552\n","16-Jun-2025 00:28:09 DEBUG GEOparse - SAMPLE: GSM2062553\n","DEBUG:GEOparse:SAMPLE: GSM2062553\n","16-Jun-2025 00:28:09 DEBUG GEOparse - SAMPLE: GSM2062554\n","DEBUG:GEOparse:SAMPLE: GSM2062554\n","16-Jun-2025 00:28:09 DEBUG GEOparse - SAMPLE: GSM2062555\n","DEBUG:GEOparse:SAMPLE: GSM2062555\n","16-Jun-2025 00:28:09 DEBUG GEOparse - SAMPLE: GSM2062556\n","DEBUG:GEOparse:SAMPLE: GSM2062556\n","16-Jun-2025 00:28:09 DEBUG GEOparse - SAMPLE: GSM2062557\n","DEBUG:GEOparse:SAMPLE: GSM2062557\n","16-Jun-2025 00:28:10 DEBUG GEOparse - SAMPLE: GSM2062558\n","DEBUG:GEOparse:SAMPLE: GSM2062558\n","16-Jun-2025 00:28:10 DEBUG GEOparse - SAMPLE: GSM2062559\n","DEBUG:GEOparse:SAMPLE: GSM2062559\n","16-Jun-2025 00:28:10 DEBUG GEOparse - SAMPLE: GSM2062560\n","DEBUG:GEOparse:SAMPLE: GSM2062560\n","16-Jun-2025 00:28:10 DEBUG GEOparse - SAMPLE: GSM2062561\n","DEBUG:GEOparse:SAMPLE: GSM2062561\n","16-Jun-2025 00:28:10 DEBUG GEOparse - SAMPLE: GSM2062562\n","DEBUG:GEOparse:SAMPLE: GSM2062562\n","16-Jun-2025 00:28:10 DEBUG GEOparse - SAMPLE: GSM2062563\n","DEBUG:GEOparse:SAMPLE: GSM2062563\n","16-Jun-2025 00:28:10 DEBUG GEOparse - SAMPLE: GSM2062564\n","DEBUG:GEOparse:SAMPLE: GSM2062564\n","16-Jun-2025 00:28:10 DEBUG GEOparse - SAMPLE: GSM2062565\n","DEBUG:GEOparse:SAMPLE: GSM2062565\n","16-Jun-2025 00:28:10 DEBUG GEOparse - SAMPLE: GSM2062566\n","DEBUG:GEOparse:SAMPLE: GSM2062566\n","16-Jun-2025 00:28:10 DEBUG GEOparse - SAMPLE: GSM2062567\n","DEBUG:GEOparse:SAMPLE: GSM2062567\n","16-Jun-2025 00:28:10 DEBUG GEOparse - SAMPLE: GSM2062568\n","DEBUG:GEOparse:SAMPLE: GSM2062568\n","16-Jun-2025 00:28:11 DEBUG GEOparse - SAMPLE: GSM2062569\n","DEBUG:GEOparse:SAMPLE: GSM2062569\n","16-Jun-2025 00:28:11 DEBUG GEOparse - SAMPLE: GSM2062570\n","DEBUG:GEOparse:SAMPLE: GSM2062570\n","16-Jun-2025 00:28:11 DEBUG GEOparse - SAMPLE: GSM2062571\n","DEBUG:GEOparse:SAMPLE: GSM2062571\n","16-Jun-2025 00:28:11 DEBUG GEOparse - SAMPLE: GSM2062572\n","DEBUG:GEOparse:SAMPLE: GSM2062572\n","16-Jun-2025 00:28:11 DEBUG GEOparse - SAMPLE: GSM2062573\n","DEBUG:GEOparse:SAMPLE: GSM2062573\n","16-Jun-2025 00:28:11 DEBUG GEOparse - SAMPLE: GSM2062574\n","DEBUG:GEOparse:SAMPLE: GSM2062574\n","16-Jun-2025 00:28:11 DEBUG GEOparse - SAMPLE: GSM2062575\n","DEBUG:GEOparse:SAMPLE: GSM2062575\n","16-Jun-2025 00:28:11 DEBUG GEOparse - SAMPLE: GSM2062576\n","DEBUG:GEOparse:SAMPLE: GSM2062576\n","16-Jun-2025 00:28:11 DEBUG GEOparse - SAMPLE: GSM2062577\n","DEBUG:GEOparse:SAMPLE: GSM2062577\n","16-Jun-2025 00:28:12 DEBUG GEOparse - SAMPLE: GSM2062578\n","DEBUG:GEOparse:SAMPLE: GSM2062578\n","16-Jun-2025 00:28:12 DEBUG GEOparse - SAMPLE: GSM2062579\n","DEBUG:GEOparse:SAMPLE: GSM2062579\n","16-Jun-2025 00:28:12 DEBUG GEOparse - SAMPLE: GSM2062580\n","DEBUG:GEOparse:SAMPLE: GSM2062580\n","16-Jun-2025 00:28:12 DEBUG GEOparse - SAMPLE: GSM2062581\n","DEBUG:GEOparse:SAMPLE: GSM2062581\n","16-Jun-2025 00:28:12 DEBUG GEOparse - SAMPLE: GSM2062582\n","DEBUG:GEOparse:SAMPLE: GSM2062582\n","16-Jun-2025 00:28:12 DEBUG GEOparse - SAMPLE: GSM2062583\n","DEBUG:GEOparse:SAMPLE: GSM2062583\n","16-Jun-2025 00:28:12 DEBUG GEOparse - SAMPLE: GSM2062584\n","DEBUG:GEOparse:SAMPLE: GSM2062584\n","16-Jun-2025 00:28:13 DEBUG GEOparse - SAMPLE: GSM2062585\n","DEBUG:GEOparse:SAMPLE: GSM2062585\n","16-Jun-2025 00:28:13 DEBUG GEOparse - SAMPLE: GSM2062586\n","DEBUG:GEOparse:SAMPLE: GSM2062586\n","16-Jun-2025 00:28:13 DEBUG GEOparse - SAMPLE: GSM2062587\n","DEBUG:GEOparse:SAMPLE: GSM2062587\n","16-Jun-2025 00:28:13 DEBUG GEOparse - SAMPLE: GSM2062588\n","DEBUG:GEOparse:SAMPLE: GSM2062588\n","16-Jun-2025 00:28:13 DEBUG GEOparse - SAMPLE: GSM2062589\n","DEBUG:GEOparse:SAMPLE: GSM2062589\n","16-Jun-2025 00:28:13 DEBUG GEOparse - SAMPLE: GSM2062590\n","DEBUG:GEOparse:SAMPLE: GSM2062590\n","16-Jun-2025 00:28:13 DEBUG GEOparse - SAMPLE: GSM2062591\n","DEBUG:GEOparse:SAMPLE: GSM2062591\n","16-Jun-2025 00:28:13 DEBUG GEOparse - SAMPLE: GSM2062592\n","DEBUG:GEOparse:SAMPLE: GSM2062592\n","16-Jun-2025 00:28:13 DEBUG GEOparse - SAMPLE: GSM2062593\n","DEBUG:GEOparse:SAMPLE: GSM2062593\n","16-Jun-2025 00:28:13 DEBUG GEOparse - SAMPLE: GSM2062594\n","DEBUG:GEOparse:SAMPLE: GSM2062594\n","16-Jun-2025 00:28:14 DEBUG GEOparse - SAMPLE: GSM2062595\n","DEBUG:GEOparse:SAMPLE: GSM2062595\n","16-Jun-2025 00:28:14 DEBUG GEOparse - SAMPLE: GSM2062596\n","DEBUG:GEOparse:SAMPLE: GSM2062596\n","16-Jun-2025 00:28:14 DEBUG GEOparse - SAMPLE: GSM2062597\n","DEBUG:GEOparse:SAMPLE: GSM2062597\n","16-Jun-2025 00:28:14 DEBUG GEOparse - SAMPLE: GSM2062598\n","DEBUG:GEOparse:SAMPLE: GSM2062598\n","16-Jun-2025 00:28:14 DEBUG GEOparse - SAMPLE: GSM2062599\n","DEBUG:GEOparse:SAMPLE: GSM2062599\n","16-Jun-2025 00:28:14 DEBUG GEOparse - SAMPLE: GSM2062600\n","DEBUG:GEOparse:SAMPLE: GSM2062600\n","16-Jun-2025 00:28:14 DEBUG GEOparse - SAMPLE: GSM2062601\n","DEBUG:GEOparse:SAMPLE: GSM2062601\n","16-Jun-2025 00:28:14 DEBUG GEOparse - SAMPLE: GSM2062602\n","DEBUG:GEOparse:SAMPLE: GSM2062602\n","16-Jun-2025 00:28:14 DEBUG GEOparse - SAMPLE: GSM2062603\n","DEBUG:GEOparse:SAMPLE: GSM2062603\n","16-Jun-2025 00:28:15 DEBUG GEOparse - SAMPLE: GSM2062604\n","DEBUG:GEOparse:SAMPLE: GSM2062604\n","16-Jun-2025 00:28:15 DEBUG GEOparse - SAMPLE: GSM2062605\n","DEBUG:GEOparse:SAMPLE: GSM2062605\n","16-Jun-2025 00:28:15 DEBUG GEOparse - SAMPLE: GSM2062606\n","DEBUG:GEOparse:SAMPLE: GSM2062606\n","16-Jun-2025 00:28:15 DEBUG GEOparse - SAMPLE: GSM2062607\n","DEBUG:GEOparse:SAMPLE: GSM2062607\n","16-Jun-2025 00:28:15 DEBUG GEOparse - SAMPLE: GSM2062608\n","DEBUG:GEOparse:SAMPLE: GSM2062608\n","16-Jun-2025 00:28:15 DEBUG GEOparse - SAMPLE: GSM2062609\n","DEBUG:GEOparse:SAMPLE: GSM2062609\n","16-Jun-2025 00:28:15 DEBUG GEOparse - SAMPLE: GSM2062610\n","DEBUG:GEOparse:SAMPLE: GSM2062610\n","16-Jun-2025 00:28:15 DEBUG GEOparse - SAMPLE: GSM2062611\n","DEBUG:GEOparse:SAMPLE: GSM2062611\n","16-Jun-2025 00:28:15 DEBUG GEOparse - SAMPLE: GSM2062612\n","DEBUG:GEOparse:SAMPLE: GSM2062612\n","16-Jun-2025 00:28:15 DEBUG GEOparse - SAMPLE: GSM2062613\n","DEBUG:GEOparse:SAMPLE: GSM2062613\n","16-Jun-2025 00:28:15 DEBUG GEOparse - SAMPLE: GSM2062614\n","DEBUG:GEOparse:SAMPLE: GSM2062614\n","16-Jun-2025 00:28:15 DEBUG GEOparse - SAMPLE: GSM2062615\n","DEBUG:GEOparse:SAMPLE: GSM2062615\n","16-Jun-2025 00:28:15 DEBUG GEOparse - SAMPLE: GSM2062616\n","DEBUG:GEOparse:SAMPLE: GSM2062616\n","16-Jun-2025 00:28:15 DEBUG GEOparse - SAMPLE: GSM2062617\n","DEBUG:GEOparse:SAMPLE: GSM2062617\n"]}]},{"cell_type":"code","source":["metadata_df"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":770},"id":"0eZFOLWCIXZz","executionInfo":{"status":"ok","timestamp":1750033825441,"user_tz":-540,"elapsed":151,"user":{"displayName":"권혜빈","userId":"01947391368488800076"}},"outputId":"6a4e909d-e53a-42c2-a526-820b7ddc1cd4"},"execution_count":24,"outputs":[{"output_type":"execute_result","data":{"text/plain":["         gsm_id subject_id                                tissue  \\\n","0    GSM2062466          1  abdominal subcutaneous white adipose   \n","1    GSM2062467          1  abdominal subcutaneous white adipose   \n","2    GSM2062468          1  abdominal subcutaneous white adipose   \n","3    GSM2062469          2  abdominal subcutaneous white adipose   \n","4    GSM2062470          2  abdominal subcutaneous white adipose   \n","..          ...        ...                                   ...   \n","147  GSM2062613         56  abdominal subcutaneous white adipose   \n","148  GSM2062614         59  abdominal subcutaneous white adipose   \n","149  GSM2062615         59  abdominal subcutaneous white adipose   \n","150  GSM2062616         61  abdominal subcutaneous white adipose   \n","151  GSM2062617         61  abdominal subcutaneous white adipose   \n","\n","                 treatment                  time_point     sex age_yrs  \\\n","0    very-low-calorie diet              at study start  female      46   \n","1    very-low-calorie diet    after weight loss period  female      46   \n","2    very-low-calorie diet  after weight stable period  female      46   \n","3    very-low-calorie diet              at study start  female      45   \n","4    very-low-calorie diet    after weight loss period  female      45   \n","..                     ...                         ...     ...     ...   \n","147       low-calorie diet  after weight stable period    male      56   \n","148       low-calorie diet              at study start    male      67   \n","149       low-calorie diet    after weight loss period    male      67   \n","150  very-low-calorie diet              at study start    male      52   \n","151  very-low-calorie diet  after weight stable period    male      52   \n","\n","    height_cm weight_kg bmi_kg/m2 body_fat_%  \n","0         167     83.38      29.9       39.9  \n","1         167     75.16     26.95       35.8  \n","2         167     73.79     26.46       32.4  \n","3         164     80.94     30.09       46.7  \n","4         164     71.92     26.74         43  \n","..        ...       ...       ...        ...  \n","147       171     89.98     30.77       36.5  \n","148     168.5     82.42     29.03       28.2  \n","149     168.5     75.01     26.42       21.2  \n","150       182     94.26     28.46       31.1  \n","151       182      88.5     26.72       26.6  \n","\n","[152 rows x 11 columns]"],"text/html":["\n","  <div id=\"df-01742c7d-2bb9-49a8-91b2-79e738de6512\" class=\"colab-df-container\">\n","    <div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>gsm_id</th>\n","      <th>subject_id</th>\n","      <th>tissue</th>\n","      <th>treatment</th>\n","      <th>time_point</th>\n","      <th>sex</th>\n","      <th>age_yrs</th>\n","      <th>height_cm</th>\n","      <th>weight_kg</th>\n","      <th>bmi_kg/m2</th>\n","      <th>body_fat_%</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>GSM2062466</td>\n","      <td>1</td>\n","      <td>abdominal subcutaneous white adipose</td>\n","      <td>very-low-calorie diet</td>\n","      <td>at study start</td>\n","      <td>female</td>\n","      <td>46</td>\n","      <td>167</td>\n","      <td>83.38</td>\n","      <td>29.9</td>\n","      <td>39.9</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>GSM2062467</td>\n","      <td>1</td>\n","      <td>abdominal subcutaneous white adipose</td>\n","      <td>very-low-calorie diet</td>\n","      <td>after weight loss period</td>\n","      <td>female</td>\n","      <td>46</td>\n","      <td>167</td>\n","      <td>75.16</td>\n","      <td>26.95</td>\n","      <td>35.8</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>GSM2062468</td>\n","      <td>1</td>\n","      <td>abdominal subcutaneous white adipose</td>\n","      <td>very-low-calorie diet</td>\n","      <td>after weight stable period</td>\n","      <td>female</td>\n","      <td>46</td>\n","      <td>167</td>\n","      <td>73.79</td>\n","      <td>26.46</td>\n","      <td>32.4</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>GSM2062469</td>\n","      <td>2</td>\n","      <td>abdominal subcutaneous white adipose</td>\n","      <td>very-low-calorie diet</td>\n","      <td>at study start</td>\n","      <td>female</td>\n","      <td>45</td>\n","      <td>164</td>\n","      <td>80.94</td>\n","      <td>30.09</td>\n","      <td>46.7</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>GSM2062470</td>\n","      <td>2</td>\n","      <td>abdominal subcutaneous white adipose</td>\n","      <td>very-low-calorie diet</td>\n","      <td>after weight loss period</td>\n","      <td>female</td>\n","      <td>45</td>\n","      <td>164</td>\n","      <td>71.92</td>\n","      <td>26.74</td>\n","      <td>43</td>\n","    </tr>\n","    <tr>\n","      <th>...</th>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","    </tr>\n","    <tr>\n","      <th>147</th>\n","      <td>GSM2062613</td>\n","      <td>56</td>\n","      <td>abdominal subcutaneous white adipose</td>\n","      <td>low-calorie diet</td>\n","      <td>after weight stable period</td>\n","      <td>male</td>\n","      <td>56</td>\n","      <td>171</td>\n","      <td>89.98</td>\n","      <td>30.77</td>\n","      <td>36.5</td>\n","    </tr>\n","    <tr>\n","      <th>148</th>\n","      <td>GSM2062614</td>\n","      <td>59</td>\n","      <td>abdominal subcutaneous white adipose</td>\n","      <td>low-calorie diet</td>\n","      <td>at study start</td>\n","      <td>male</td>\n","      <td>67</td>\n","      <td>168.5</td>\n","      <td>82.42</td>\n","      <td>29.03</td>\n","      <td>28.2</td>\n","    </tr>\n","    <tr>\n","      <th>149</th>\n","      <td>GSM2062615</td>\n","      <td>59</td>\n","      <td>abdominal subcutaneous white adipose</td>\n","      <td>low-calorie diet</td>\n","      <td>after weight loss period</td>\n","      <td>male</td>\n","      <td>67</td>\n","      <td>168.5</td>\n","      <td>75.01</td>\n","      <td>26.42</td>\n","      <td>21.2</td>\n","    </tr>\n","    <tr>\n","      <th>150</th>\n","      <td>GSM2062616</td>\n","      <td>61</td>\n","      <td>abdominal subcutaneous white adipose</td>\n","      <td>very-low-calorie diet</td>\n","      <td>at study start</td>\n","      <td>male</td>\n","      <td>52</td>\n","      <td>182</td>\n","      <td>94.26</td>\n","      <td>28.46</td>\n","      <td>31.1</td>\n","    </tr>\n","    <tr>\n","      <th>151</th>\n","      <td>GSM2062617</td>\n","      <td>61</td>\n","      <td>abdominal subcutaneous white adipose</td>\n","      <td>very-low-calorie diet</td>\n","      <td>after weight stable period</td>\n","      <td>male</td>\n","      <td>52</td>\n","      <td>182</td>\n","      <td>88.5</td>\n","      <td>26.72</td>\n","      <td>26.6</td>\n","    </tr>\n","  </tbody>\n","</table>\n","<p>152 rows × 11 columns</p>\n","</div>\n","    <div class=\"colab-df-buttons\">\n","\n","  <div class=\"colab-df-container\">\n","    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-01742c7d-2bb9-49a8-91b2-79e738de6512')\"\n","            title=\"Convert this dataframe to an interactive table.\"\n","            style=\"display:none;\">\n","\n","  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n","    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n","  </svg>\n","    </button>\n","\n","  <style>\n","    .colab-df-container {\n","      display:flex;\n","      gap: 12px;\n","    }\n","\n","    .colab-df-convert {\n","      background-color: #E8F0FE;\n","      border: none;\n","      border-radius: 50%;\n","      cursor: pointer;\n","      display: none;\n","      fill: #1967D2;\n","      height: 32px;\n","      padding: 0 0 0 0;\n","      width: 32px;\n","    }\n","\n","    .colab-df-convert:hover {\n","      background-color: #E2EBFA;\n","      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n","      fill: #174EA6;\n","    }\n","\n","    .colab-df-buttons div {\n","      margin-bottom: 4px;\n","    }\n","\n","    [theme=dark] .colab-df-convert {\n","      background-color: #3B4455;\n","      fill: #D2E3FC;\n","    }\n","\n","    [theme=dark] .colab-df-convert:hover {\n","      background-color: #434B5C;\n","      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n","      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n","      fill: #FFFFFF;\n","    }\n","  </style>\n","\n","    <script>\n","      const buttonEl =\n","        document.querySelector('#df-01742c7d-2bb9-49a8-91b2-79e738de6512 button.colab-df-convert');\n","      buttonEl.style.display =\n","        google.colab.kernel.accessAllowed ? 'block' : 'none';\n","\n","      async function convertToInteractive(key) {\n","        const element = document.querySelector('#df-01742c7d-2bb9-49a8-91b2-79e738de6512');\n","        const dataTable =\n","          await google.colab.kernel.invokeFunction('convertToInteractive',\n","                                                    [key], {});\n","        if (!dataTable) return;\n","\n","        const docLinkHtml = 'Like what you see? Visit the ' +\n","          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n","          + ' to learn more about interactive tables.';\n","        element.innerHTML = '';\n","        dataTable['output_type'] = 'display_data';\n","        await google.colab.output.renderOutput(dataTable, element);\n","        const docLink = document.createElement('div');\n","        docLink.innerHTML = docLinkHtml;\n","        element.appendChild(docLink);\n","      }\n","    </script>\n","  </div>\n","\n","\n","    <div id=\"df-a9e550af-3071-48d4-9645-245afac4e948\">\n","      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-a9e550af-3071-48d4-9645-245afac4e948')\"\n","                title=\"Suggest charts\"\n","                style=\"display:none;\">\n","\n","<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n","     width=\"24px\">\n","    <g>\n","        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n","    </g>\n","</svg>\n","      </button>\n","\n","<style>\n","  .colab-df-quickchart {\n","      --bg-color: #E8F0FE;\n","      --fill-color: #1967D2;\n","      --hover-bg-color: #E2EBFA;\n","      --hover-fill-color: #174EA6;\n","      --disabled-fill-color: #AAA;\n","      --disabled-bg-color: #DDD;\n","  }\n","\n","  [theme=dark] .colab-df-quickchart {\n","      --bg-color: #3B4455;\n","      --fill-color: #D2E3FC;\n","      --hover-bg-color: #434B5C;\n","      --hover-fill-color: #FFFFFF;\n","      --disabled-bg-color: #3B4455;\n","      --disabled-fill-color: #666;\n","  }\n","\n","  .colab-df-quickchart {\n","    background-color: var(--bg-color);\n","    border: none;\n","    border-radius: 50%;\n","    cursor: pointer;\n","    display: none;\n","    fill: var(--fill-color);\n","    height: 32px;\n","    padding: 0;\n","    width: 32px;\n","  }\n","\n","  .colab-df-quickchart:hover {\n","    background-color: var(--hover-bg-color);\n","    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n","    fill: var(--button-hover-fill-color);\n","  }\n","\n","  .colab-df-quickchart-complete:disabled,\n","  .colab-df-quickchart-complete:disabled:hover {\n","    background-color: var(--disabled-bg-color);\n","    fill: var(--disabled-fill-color);\n","    box-shadow: none;\n","  }\n","\n","  .colab-df-spinner {\n","    border: 2px solid var(--fill-color);\n","    border-color: transparent;\n","    border-bottom-color: var(--fill-color);\n","    animation:\n","      spin 1s steps(1) infinite;\n","  }\n","\n","  @keyframes spin {\n","    0% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","      border-left-color: var(--fill-color);\n","    }\n","    20% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    30% {\n","      border-color: transparent;\n","      border-left-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","      border-right-color: var(--fill-color);\n","    }\n","    40% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-top-color: var(--fill-color);\n","    }\n","    60% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","    }\n","    80% {\n","      border-color: transparent;\n","      border-right-color: var(--fill-color);\n","      border-bottom-color: var(--fill-color);\n","    }\n","    90% {\n","      border-color: transparent;\n","      border-bottom-color: var(--fill-color);\n","    }\n","  }\n","</style>\n","\n","      <script>\n","        async function quickchart(key) {\n","          const quickchartButtonEl =\n","            document.querySelector('#' + key + ' button');\n","          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n","          quickchartButtonEl.classList.add('colab-df-spinner');\n","          try {\n","            const charts = await google.colab.kernel.invokeFunction(\n","                'suggestCharts', [key], {});\n","          } catch (error) {\n","            console.error('Error during call to suggestCharts:', error);\n","          }\n","          quickchartButtonEl.classList.remove('colab-df-spinner');\n","          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n","        }\n","        (() => {\n","          let quickchartButtonEl =\n","            document.querySelector('#df-a9e550af-3071-48d4-9645-245afac4e948 button');\n","          quickchartButtonEl.style.display =\n","            google.colab.kernel.accessAllowed ? 'block' : 'none';\n","        })();\n","      </script>\n","    </div>\n","\n","  <div id=\"id_ace79a9c-30c1-469d-bca1-54739fcb24f8\">\n","    <style>\n","      .colab-df-generate {\n","        background-color: #E8F0FE;\n","        border: none;\n","        border-radius: 50%;\n","        cursor: pointer;\n","        display: none;\n","        fill: #1967D2;\n","        height: 32px;\n","        padding: 0 0 0 0;\n","        width: 32px;\n","      }\n","\n","      .colab-df-generate:hover {\n","        background-color: #E2EBFA;\n","        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n","        fill: #174EA6;\n","      }\n","\n","      [theme=dark] .colab-df-generate {\n","        background-color: #3B4455;\n","        fill: #D2E3FC;\n","      }\n","\n","      [theme=dark] .colab-df-generate:hover {\n","        background-color: #434B5C;\n","        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n","        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n","        fill: #FFFFFF;\n","      }\n","    </style>\n","    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('metadata_df')\"\n","            title=\"Generate code using this dataframe.\"\n","            style=\"display:none;\">\n","\n","  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n","       width=\"24px\">\n","    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n","  </svg>\n","    </button>\n","    <script>\n","      (() => {\n","      const buttonEl =\n","        document.querySelector('#id_ace79a9c-30c1-469d-bca1-54739fcb24f8 button.colab-df-generate');\n","      buttonEl.style.display =\n","        google.colab.kernel.accessAllowed ? 'block' : 'none';\n","\n","      buttonEl.onclick = () => {\n","        google.colab.notebook.generateWithVariable('metadata_df');\n","      }\n","      })();\n","    </script>\n","  </div>\n","\n","    </div>\n","  </div>\n"],"application/vnd.google.colaboratory.intrinsic+json":{"type":"dataframe","variable_name":"metadata_df","summary":"{\n  \"name\": \"metadata_df\",\n  \"rows\": 152,\n  \"fields\": [\n    {\n      \"column\": \"gsm_id\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 152,\n        \"samples\": [\n          \"GSM2062534\",\n          \"GSM2062613\",\n          \"GSM2062562\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"subject_id\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 53,\n        \"samples\": [\n          \"23\",\n          \"47\",\n          \"53\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"tissue\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"abdominal subcutaneous white adipose\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"treatment\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 2,\n        \"samples\": [\n          \"low-calorie diet\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"time_point\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 3,\n        \"samples\": [\n          \"at study start\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"sex\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 2,\n        \"samples\": [\n          \"male\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"age_yrs\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 27,\n        \"samples\": [\n          \"40\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"height_cm\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 36,\n        \"samples\": [\n          \"171\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"weight_kg\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 148,\n        \"samples\": [\n          \"99.32\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"bmi_kg/m2\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 142,\n        \"samples\": [\n          \"27.79\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"body_fat_%\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 129,\n        \"samples\": [\n          \"48.1\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"}},"metadata":{},"execution_count":24}]},{"cell_type":"code","source":["metadata_df.to_csv(\"/content/drive/MyDrive/BIO_MEDICAL_PROJECT/obesity_gene_analysis/data/metadata_df.csv\")"],"metadata":{"id":"SZfJ-_z_JStD","executionInfo":{"status":"ok","timestamp":1750033861921,"user_tz":-540,"elapsed":14,"user":{"displayName":"권혜빈","userId":"01947391368488800076"}}},"execution_count":25,"outputs":[]},{"cell_type":"markdown","source":["### * 비만 관련 유전자 개수 몇 개?\n","GWAS Catalog, OMIM, DisGeNET 등에서 **비만(BMI, waist circumference, obesity 등)**과 관련 있다고 보고된 유전자는 수백 개 이상.\n","\n","\n","### * 10개 정도만 골라도 괜찮은 이유\n","\n","| 이유                   | 설명                                                                                        |\n","| -------------------- | ----------------------------------------------------------------------------------------- |\n","| **1. 대표성 있는 상위 유전자** | FTO, MC4R, LEP 등은 **거의 모든 GWAS에서 등장**하는 핵심 유전자입니다.                                        |\n","| **2. 예측 모델의 목적**     | 정량적 해석(XAI 기반 SHAP)과 **설명 가능성 확보**를 목표로 할 경우, **적은 수의 강력한 feature**를 쓰는 것이 오히려 해석에 유리합니다. |\n","| **3. 학습 데이터 크기**     | 현재 subject 수가 53명이라 너무 많은 유전자를 쓰면 오히려 과적합 위험이 있습니다.                                       |\n","\n","### * non-obesity 유전자 발굴 시 정확성\n","\n","SHAP은 실제로 **각 유전자 발현량과 감량량/저항성**의 상관성을 통해 영향을 분석\n","따라서 비만 관련 여부는 SHAP은 모름. 이건 나중에 해석할 때 연구자가 확인하는 것. SHAP 상위 유전자 중 비만유전자 아닌 것 추출\t“잠재적 신규 비만 예측 유전자” 후보→ 논문 조사, KEGG pathway 매핑\t기능 해석을 통한 생물학적 의미 부여|\n"],"metadata":{"id":"mC7bU9hFE0tI"}},{"cell_type":"markdown","source":["### * 비만 유전자 12개 요약 표\n","\n","| 유전자명       | 기능 설명                                                | 주요 경로                                     | 대사 관련 역할                             | 대표 논문 (PMID)                                               |\n","| ---------- | ---------------------------------------------------- | ----------------------------------------- | ------------------------------------ | ---------------------------------------------------------- |\n","| **FTO**    | **m6A RNA demethylase**. 에너지 섭취 조절, 지방세포 분화          | **mTOR, AMPK, GH/IGF-1**                  | 식욕 증가, 칼로리 섭취 증가, 체중 증가 유도           | [PMID: 17632544](https://pubmed.ncbi.nlm.nih.gov/17632544) |\n","| **MC4R**   | **Melanocortin 4 receptor**. 시상하부에서 식욕 억제 역할         | **Leptin-MSH pathway**                    | 식욕 억제 → 체중 유지. 돌연변이 시 비만 발생          | [PMID: 12538646](https://pubmed.ncbi.nlm.nih.gov/12538646) |\n","| **LEP**    | **Leptin 단백질** 생성. 지방세포에서 분비되어 식욕 억제 신호 전달           | **JAK/STAT, PI3K-AKT**                    | 포만감 유도, 에너지 항상성. 렙틴 결핍 시 병적 비만       | [PMID: 8552186](https://pubmed.ncbi.nlm.nih.gov/8552186)   |\n","| **LEPR**   | **Leptin receptor**. 시상하부에서 leptin 신호 수용             | **JAK2/STAT3, IRS**                       | 렙틴 저항성 발생 시 체중 증가 → 비만 유발            | [PMID: 9435158](https://pubmed.ncbi.nlm.nih.gov/9435158)   |\n","| **ADIPOQ** | **Adiponectin** 단백질 생성. 항염증성, 인슐린 감수성 증가             | **AMPK, PPAR-α**                          | 체지방 감소 촉진, 인슐린 저항성 개선, 대사 증후군 보호 인자  | [PMID: 11240151](https://pubmed.ncbi.nlm.nih.gov/11240151) |\n","| **PPARG**  | **PPARγ nuclear receptor**. 지방세포 분화 조절               | **Adipogenesis, lipid metabolism**        | 지방축적 조절, 티아졸리디온계 약물의 표적 → 당뇨 및 비만 치료 | [PMID: 11099450](https://pubmed.ncbi.nlm.nih.gov/11099450) |\n","| **UCP1**   | **Mitochondrial uncoupling protein**. 갈색 지방에서 열 생성   | **Thermogenesis pathway**                 | **열 발생 통한 지방 연소** → 에너지 소비 증가        | [PMID: 8197456](https://pubmed.ncbi.nlm.nih.gov/8197456)   |\n","| **UCP2**   | UCP1과 유사하지만 **백색지방/췌장 등 다양한 조직에 존재**                 | **ROS 억제, ATP 생산 조절**                     | 에너지 효율 감소, 비만/당뇨 등 대사 질환 연관          | [PMID: 10888871](https://pubmed.ncbi.nlm.nih.gov/10888871) |\n","| **BDNF**   | **Brain-Derived Neurotrophic Factor**. 뇌에서 식욕 조절에 영향 | **TrkB-NPY, hypothalamus signaling**      | 식욕 억제, BDNF 결핍 시 과식 유도               | [PMID: 10471511](https://pubmed.ncbi.nlm.nih.gov/10471511) |\n","| **GNPDA2** | **Glucosamine-6-phosphate isomerase**. 신경조직에서 발현     | **Hexosamine biosynthetic pathway**       | GWAS에서 비만 관련 다수 연관 보고됨 (기전은 불명확)     | [PMID: 18953394](https://pubmed.ncbi.nlm.nih.gov/18953394) |\n","| **TMEM18** | **Transmembrane protein 18**. 시상하부에서 발현, 세포 이동과 관련   | **Neuronal regulation**                   | GWAS에서 체질량 지수(BMI)와 강한 상관관계          | [PMID: 20031578](https://pubmed.ncbi.nlm.nih.gov/20031578) |\n","| **NEGR1**  | **Neuronal Growth Regulator 1**. 신경 발달 및 식욕 조절       | **Neural plasticity, appetite signaling** | 뇌 기능 변화 통한 식욕 조절. GWAS 상위 유전자        | [PMID: 19079260](https://pubmed.ncbi.nlm.nih.gov/19079260) |\n"],"metadata":{"id":"k8ux-Cs-Gt-1"}}]}